% THIS IS CURRENTLY A ROUGH DRAFT
% QUESTION: How do I deal with experiment repeats?  Do I pile the data together, or do I put them in the appendix?
% TODO:
% - Bullet points -> prose
% - Notes/random ideas jotted down -> comments

\chapter{Microfluidics and fluorescence microscopy for cellular metabolic cycles}
\label{ch:biology}

There is a paucity of studies of the yeast metabolic cycle at the cellular level --- i.e.\ in which budding yeast cells are investigated in isolation rather than as a population.
Previous population-level studies in the chemostat, by their very nature, fail to account for cell-to-cell heterogeneity and create cell density and environmental conditions that are far removed from the natural habitat of budding yeast.

Here, I use single-cell microfluidics to physically separate budding yeast cells and use fluorescence microscopy to monitor the yeast metabolic cycle and the cell division cycle.
I do so in order to address the relationship between the metabolic and cell division cycle, and whether nutrient and genetic perturbations affect the metabolic cycle as was previously reported in chemostat-based studies.
Specifically, I culture wild-type and engineered strains, including deletion strains, in microfluidic chambers that force the cells to physically separate.
Each cell in a trap is monitored while it undergoes growth and cell division: brightfield images and images of flavin fluorescence are taken.
In some experiments, I use the microfluidics set-up to create rapid changes in the nutrient medium at specified times to assess the cells' response.

My aims in this chapter are:
\begin{enumerate}
  \item To confirm that flavin autofluorescence can be used to monitor the yeast metabolic cycle \parencite{baumgartnerFlavinbasedMetabolicCycles2018}, and to confirm autonomy of the metabolic oscillator from the cell division oscillator.
  \item To show the robustness of the yeast metabolic cycle across nutrient conditions and deletion strains.
  \item To address whether observations of the behaviour of the metabolic cycles of deletion strains in the chemostat can be recapitulated in single-cell conditions.
        And, if behaviours differ, offer an explanation based on cell-to-cell heterogeneity.
\end{enumerate}

\section[Permissive conditions]{Single-cell flavin oscillations synchronise with cell division cycle in permissive conditions.}
\label{sec:biology-sync}

\begin{figure}
  \centering
    \includegraphics[width=0.5\linewidth]{garmendia-torresMultipleInputsEnsure2018_1_adapted.jpg}
    \caption{
      Engineering a fluorescent protein cassette fused to \textit{HTB2} (A) allows the identification of phases of the cell division cycle by modelling the fluorescent protein intensity changes over time as piecewise linear sections (B).
      Adapted from \textcite{garmendia-torresMultipleInputsEnsure2018}.
    }
  \label{fig:biology-htb2}
\end{figure}

To replicate the single-cell flavin oscillations observed by \textcite{baumgartnerFlavinbasedMetabolicCycles2018}, I cultured FY4 HTB2::mCherry cells in minimal medium supplemented with \SI{20}{\gram~\litre^{-1}} glucose, over \SI{24}{\hour} in the microfluidic device.
These cells are prototrophic, owing to its FY4 background, allowing me to rule out the effects of certain nutrient supplements on the metabolic cycle.
I engineered an insertion to the \textit{HTB2} gene to attach a fluorescent protein to histone 2B so that imaging of the fluorescence protein shows phases of the cell division cycle, as performed by ~\textcite{garmendia-torresMultipleInputsEnsure2018} (figure~\ref{fig:biology-htb2}).
I chose mCherry as the fluorescent protein because the GFP emission spectrum overlaps with the flavin emission spectrum, and because mutual information does not show a significant difference between GFP and superfolder GFP traces (figure ...).
Finally, I chose the high glucose concentration \SI{20}{\gram~\litre^{-1}} glucose as it is commonly used as a permissive growth condition for budding yeast cultivation experiments.

Replicating a previous flavin-based microfluidics study to confirm that my experimental set-up monitors the yeast metabolic cycle is important.
This is because I use a different single-cell microfluidics setup from previous studies and we use flavin, a fluorescent compound that is less used as an indicator of the yeast metabolic cycle.
In addition, I aim to show the use of a more informative cell division cycle marker, compared to the more commonly used Whi5p, along with automated identification of budding events using \textit{BABY} \parencite{pietschDeterminingGrowthRates2023}.


\begin{figure}
  \centering
    \includegraphics[width=1.0\linewidth]{single_birth_plot_edit.pdf}
    \caption{
      Oscillations of flavin fluorescence in the yeast metabolic cycle synchronise with events in the cell division cycle.
      Figure shows sample flavin fluorescence (purple, solid lines) and histone 2B (pink, dotted lines) level in a single FY4 HTB2::mCherry cell grown in \SI{20}{\gram~\litre^{-1}} glucose.
      Vertical lines (black, dashed) indicate budding.
    }
  \label{fig:biology-highglc-single}
\end{figure}

In this experiment, I observed oscillations in the fluorescence of flavins over time from single cells (figure~\ref{fig:biology-highglc-single}).
For a given cell, these oscillations peak when a bud forms, approximately during G2/M.
This relationship between flavin oscillations and the cell division cycle indeed agrees with \textcite{baumgartnerFlavinbasedMetabolicCycles2018}.
% ... elaborate further.  Do you expect this relationship, and why?  Use biological knowledge to explain.
However, there were some cases in which there was an oscillation without cell division cycle progression or bud formation (figure ...).
Such cases, also revealed by \textcite{papagiannakisAutonomousMetabolicOscillations2017} via NAD(P)H cycles, confirm that the metabolic cycle is generated autonomously from the cell division cycle.


\begin{figure}
  \centering
  \begin{subfigure}[htpb]{0.4\textwidth}
   \centering
   \includegraphics[width=\textwidth]{htb2mCherry_26643_14}
   \caption{
    Mean Fourier spectrum of flavin fluorescence across cells.
   }
   \label{fig:biology-highglc-sync-fourier}
  \end{subfigure}%
  \begin{subfigure}[htpb]{0.4\textwidth}
   \centering
   \includegraphics[width=\textwidth]{htb2mCherry_26643_12}
   \caption{
     \textbf{(Main)} Median autocorrelation functions of flavin fluorescence (blue) and histone 2B levels (orange) time series,
     along with \textbf{(inset)} the periods of each oscillator across cells as determined by the strongest frequency in each signal's Fourier spectrum.
   }
   \label{fig:biology-highglc-sync-acf}
  \end{subfigure}

  \begin{subfigure}[htpb]{0.4\textwidth}
   \centering
   \includegraphics[width=\textwidth]{heatmap_edit.pdf}
   \caption{
     Heatmap shows the flavin fluorescence (pixels on a red-blue scale) and budding events (black pixels) of each cell.
     Signals are aligned by the first budding event.
   }
   \label{fig:biology-highglc-sync-heatmap}
  \end{subfigure}%
  \begin{subfigure}[htpb]{0.4\textwidth}
   \centering
   \includegraphics[width=\textwidth]{htb2mCherry_26643_6}
   \caption{
    Median flavin fluorescence signal across cells, aligned to first budding event.
   }
   \label{fig:biology-highglc-sync-median}
  \end{subfigure}

  \begin{subfigure}[htpb]{0.4\textwidth}
   \centering
   \includegraphics[width=\textwidth]{xcf_edit.pdf}
   \caption{
    Cross-correlation between flavin and histone 2B signals.
   }
   \label{fig:biology-highglc-sync-xcf}
  \end{subfigure}

  \caption{
    The yeast metabolic cycle and cell division cycle synchronise within each cell in a population of FY4 HTB2::mCherry cells under the same conditions (\SI{20}{\gram~\litre^{-1}} glucose).
  }
  \label{fig:biology-highglc-sync}
\end{figure}

% TODO: Add references to chapter 3 re the analysis methods
To quantify the period of the oscillators, I used several methods that confirm each other.
First, I computed the Fourier spectrum of each flavin time series, originating from each cell.
Only time series that had data points that filled all time points in the experiment were used, as the Fourier spectrum can only be calculated for such time series.
The mean of these spectra across each time series was computed (figure~\ref{fig:biology-highglc-sync-fourier}), and the peak power suggests a \SI{94.33}{\minute} period of the metabolic oscillations.
In addition, the peak power from the Fourier spectrum of each cell was determined and the distribution of such peaks was shown in a histogram to show the distribution of the periods across the population (figure~\ref{fig:biology-highglc-sync-acf} inset), additionally confirming a $\approx$\SI{90}{\minute} period.
This computation was also repeated for the mCherry time series.
Second, I computed the autocorrelation function of flavin time series and of mCherry time series across the population of cells, confirming a $\approx$\SI{90}{\minute} period for both time series (figure~\ref{fig:biology-highglc-sync-acf}).
% [CITATIONS NEEDED]
Taken together, these methods suggest that both the metabolic and cell division cycle of the prototrophic strain in high-glucose conditions occur at a period of approximately \SI{90}{\minute}, which agrees with the literature regarding the duration of the cell division cycle in such permissive growth conditions.

To visualise the relationship between the metabolic cycle and the cell division cycle, I showed the flavin fluorescence and budding events in a heatmap.
In figure~\ref{fig:biology-highglc-sync-heatmap},
each row of pixels represents a cell.
Fluorescence is represented as colours (red-blue), with red indicating low signal and blue indicating high signal.
Each budding event is indicated as a black pixel.
Signals are aligned by the first budding event to show that
(a) budding events synchronise with peaks in fluorescence, and
(b) the cell division cycle varies between cells.
The heatmap shows that most first cell division cycles --- shown as the first interval between budding events --- are just under \SI{2}{\hour}, with cell-to-cell variations, agreeing with figure~\ref{fig:biology-highglc-sync-acf} (inset).
It also shows that the budding events coincide with peaks in flavin fluorescence.
Additionally, the median of the flavin signals aligned by the first birth event is computed (figure~\ref{fig:biology-highglc-sync-median}), and the oscillatory shape of the median confirms the synchrony between the metabolic cycle and budding events.

To quantify the relationship between the metabolic cycle and the cell division cycle, I computed the cross-correlation function between the flavin and mCherry signals across the population (figure~\ref{fig:biology-highglc-sync-xcf}).
This function shows that the mCherry signals are, on average, shifted by \SI{5}{\minute} relative to the flavin signals.

Figure~\ref{fig:biology-highglc-sync} thus confirms that, consistently across a population of physically separated prototrophic cells, the metabolic cycle and the cell division cycle synchronise, with flavin intensity peaking as the cell forms a bud in G2/M phase.


% Figure & discussion of BY4741.
\begin{figure}
  \centering
  \begin{subfigure}[htpb]{0.4\textwidth}
   \centering
   \includegraphics[width=\textwidth]{by4741_491_13}
   \caption{
    Mean Fourier spectrum of flavin fluorescence across cells.
   }
   \label{fig:biology-by4741-sync-fourier}
  \end{subfigure}%
  \begin{subfigure}[htpb]{0.4\textwidth}
   \centering
   \includegraphics[width=\textwidth]{by4741_491_12}
   \caption{
     \textbf{(Main)} Median autocorrelation function flavin fluorescence,
     along with \textbf{(inset)} the periods across cells as determined by the strongest frequency in each signal's Fourier spectrum.
   }
   \label{fig:biology-by4741-sync-acf}
  \end{subfigure}

  \begin{subfigure}[htpb]{0.4\textwidth}
   \centering
   \includegraphics[width=\textwidth]{by4741_491_7.pdf}
   \caption{
    Heatmap shows the flavin fluorescence of each cell in a population over time.
    Each budding event is indicated as a black pixel.
    Signals are aligned by the first budding event.
   }
   \label{fig:biology-by4741-sync-heatmap}
  \end{subfigure}%
  \begin{subfigure}[htpb]{0.4\textwidth}
   \centering
   \includegraphics[width=\textwidth]{by4741_491_6}
   \caption{
    Median flavin fluorescence signal across cells.
   }
   \label{fig:biology-by4741-sync-median}
  \end{subfigure}

  \caption{
    Cells of the BY4741 strain consistently show an approximately 75-hour cycle of flavin oscillations.
    The presence of such oscillations in an auxotrophic strain highlights the robustness of the yeast metabolic cycle.
  }
  \label{fig:biology-by4741-sync}
\end{figure}

To show that the metabolic cycle is robust, I performed a similar experiment on an auxotrophic strain, which has genetic mutations that make it unable to grow in media lacking certain nutrients.
I cultured BY4741 cells in minimal medium supplemented with uracil and the amino acids required for this strain to grow, plus \SI{10}{\gram~\litre^{-1}} glucose.
Showing that metabolic cycles occur in an auxotroph is important because it confirms that the YMC is robust --- a lot of things must be impaired for the cycle to disappear --- and thus likely an intrinsic cycle of budding yeast.
Similar to FY4 HTB2::mCherry cells, BY4741 cells show robust, consistent oscillations in flavin fluorescence that peak upon budding (figure~\ref{fig:biology-by4741-sync}), although metabolic cycles have a period of $\approx$\SI{75}{\minute} in this case.
The shorter period, implying a higher growth rate, may be explained by a lack of burden caused by a lack of an mCherry insertion, and by media supplements.
% Discussion (probably in discussion section at the end): to be 'proper', I should perform an experiment with the BY4741 HTB2::mCherry I engineered.


\section[Abrupt nutrient changes]{Cells autonomously generate flavin oscillations, independently of the cell division cycle, in response to abrupt nutrient changes.}
\label{sec:biology-abrupt}

\begin{itemize}
\item \emph{Bottom line:} We found that when cells grown in high (7.5 g/L) glucose were abruptly starved of glucose, their flavin oscillations reset phase.  These flavin oscillations under starvation continue without budding events or with sparse budding events.
\item \emph{Importance of this section:}

\begin{itemize}
\item Many chemostat-based studies (citations needed) along with a modelling study (\textcite{krishnaMinimalPushPull2018}) suggest that a diffusible metabolite is responsible for the synchrony of the metabolic cycle across cells, while single-cell studies suggest that cells independently generate these oscillations.  I aim to confirm the conclusions from \textcite{papagiannakisAutonomousMetabolicOscillations2017}, but in flavin.
\end{itemize}
\item \emph{Evidence:}
\begin{itemize}
\item Figure 3.
\item When cells are in high glucose, oscillations are asynchronous (confirms \textcite{papagiannakisAutonomousMetabolicOscillations2017,baumgartnerFlavinbasedMetabolicCycles2018}).
\end{itemize}

\item \textbf{Missing experiments:}
\begin{itemize}
\item Adding acetate, acetaldehyde, or ethanol in bulk.  See if it synchronises YMCs, as described in chemostats by \textcite{kuangMsn2RegulateExpression2017} and \textcite{krishnaMinimalPushPull2018} .  If so, this would start a discussion about what processes trigger resetting YMC phases and their biological explanations.
\item Feast-and-famine, or glucose pulsing (\textcite{charvinForcedPeriodicExpression2009}).
\end{itemize}
\item \textbf{Missing discussion:}
\begin{itemize}
\item I observed no/weaker synchrony of oscillations between cells if the cells were grown in 10 g/L glucose.  Perhaps we can spin this as a result.  In any case, we need supplementary information to justify 7.5 g/L glucose when I used 20 g/L earlier.  This information can be Monod's Law experiments (e.g. by Hongpei) that suggests that 7.5 g/L is past saturation.
\end{itemize}
\end{itemize}

\begin{figure}
  \centering
  \begin{subfigure}[htpb]{1.0\textwidth}
   \centering
   \includegraphics[width=\textwidth]{starvation_single_birth_plot_new_edit.pdf}
   \caption{
     Flavin fluorescence synchronises with the cell division cycle in high-glucose conditions before and after starvation (FY4 HTB2::mCherry).
     %Figure shows sample flavin fluorescence levels (purple) and histone 2B localisation (pink) in single cells.
     %Vertical lines (black) indicate budding.
   }
   \label{fig:biology-starvation-single}
  \end{subfigure}

  \begin{subfigure}[htpb]{0.7\textwidth}
   \centering
   \includegraphics[width=\textwidth]{heatmap_012_edit.pdf}
   \caption{
     Heatmap shows the flavin fluorescence of each cell (FY4) in a population over time.
     %Each row of pixels represents one of 60 cells.
     %Fluorescence is represented as colours with blue indicating low signal and red indicating high signal.
     Each budding event is indicated as a black pixel.
     During starvation, synchronisation of flavin oscillations between cells were observed, and budding events were sparse.
   }
   \label{fig:biology-starvation-heatmap}
  \end{subfigure}

  \caption{
     Metabolic cycles between physically separated cells synchronise upon abrupt starvation.
     FY4 and HTB2::mCherry cells were grown in \SI{7.5}{\gram~\litre^{-1}} glucose for \SI{7}{\hour} before being abruptly switched to \SI{0}{\gram~\litre^{-1}} glucose for \SI{8}{\hour} and then resumed to \SI{7.5}{\gram~\litre^{-1}} glucose for \SI{7}{\hour}.
  }
  \label{fig:biology-starvation}
\end{figure}


\begin{figure}
  \centering
  \begin{subfigure}[htpb]{1.0\textwidth}
   \centering
   \includegraphics[width=\textwidth]{starvation_raw_13-07-02.pdf}
   \caption{
   }
   \label{fig:biology-starvation-raw-1}
  \end{subfigure}

  \begin{subfigure}[htpb]{1.0\textwidth}
   \centering
   \includegraphics[width=\textwidth]{starvation_raw_13-39-01.pdf}
   \caption{
   }
   \label{fig:biology-starvation-raw-2}
  \end{subfigure}

  \caption{
    Upon starvation, histone 2B levels can remain at a high (~\ref{fig:biology-starvation-raw-1}) or low (~\ref{fig:biology-starvation-raw-2}) level.
    Note: unlike the other figures showing single-cell time series, the flavin and mCherry time series were normalised to give zero mean and standard deviation of 1 so that they can be plot on the same vertical axes, but the high-pass Butterworth filter was not applied to emphasise the trajectories of raw values.
  }
  \label{fig:biology-starvation-raw}
\end{figure}

Insert text here.  Discussion: cell stuck in G1/G2 upon starvation.


\begin{figure}
  \centering
  \begin{subfigure}[htpb]{0.9\textwidth}
   \centering
   \includegraphics[width=\textwidth]{19972_distribs_0360}
   \caption{
     %foobar
     \SI{360}{\minute}
   }
   \label{fig:biology-starvation-distribs-0360}
  \end{subfigure}

  \begin{subfigure}[htpb]{0.9\textwidth}
   \centering
   \includegraphics[width=\textwidth]{19972_distribs_0420}
   \caption{
     \SI{420}{\minute}
   }
   \label{fig:biology-starvation-distribs-0420}
  \end{subfigure}

  \begin{subfigure}[htpb]{0.9\textwidth}
   \centering
   \includegraphics[width=\textwidth]{19972_distribs_0480}
   \caption{
     \SI{480}{\minute}
   }
   \label{fig:biology-starvation-distribs-0480}
  \end{subfigure}

  \begin{subfigure}[htpb]{0.9\textwidth}
   \centering
   \includegraphics[width=\textwidth]{19972_distribs_0540}
   \caption{
     \SI{540}{\minute}
   }
   \label{fig:biology-starvation-distribs-0540}
  \end{subfigure}

  \begin{subfigure}[htpb]{0.9\textwidth}
   \centering
   \includegraphics[width=\textwidth]{19972_distribs_0600}
   \caption{
     \SI{600}{\minute}
   }
   \label{fig:biology-starvation-distribs-0600}
  \end{subfigure}

  \begin{subfigure}[htpb]{0.9\textwidth}
   \centering
   \includegraphics[width=\textwidth]{19972_distribs_1000}
   \caption{
     \SI{1000}{\minute}
   }
   \label{fig:biology-starvation-distribs-1000}
  \end{subfigure}

  \caption{
    Distribution of quantities at selected time points in the glucose-starvation experiment, starvation being \SIrange{420}{900}{\minute}.
    HTB2::mCherry and flavin signals were based on raw data extracted from the fluorescent images.
    Growth rate was calculated based on the rolling mean of the time point-to-time point change in cell volume estimated by \emph{aliby}, and the budding probability was calculated based on the rolling mean of the boolean vector that indicates the presence or absence of budding event over time.
  }
  \label{fig:biology-starvation-distribs}
\end{figure}


\begin{figure}
  \centering
  \includegraphics[width=\textwidth]{19972_ks_highlight}
  \caption{
    Glucose-starvation experiment.
    For each quantity, the Kolmogorov-Smirnov test statistics from two-tailed tests between the distribution at \SI{360}{\minute} and the distribution at each time point of interest is shown.
    The null hypothesis for each test is that the two samples are drawn from the same probability distribution, and thick lines indicate where the hypothesis is rejected ($p < 0.05$).
    Vertical lines (red) indicate changes in nutrient conditions.
  }
  \label{fig:biology-starvation-ks}
\end{figure}


\begin{figure}
  \centering
  \begin{subfigure}[htpb]{0.7\textwidth}
   \centering
   \includegraphics[width=\textwidth]{allstrains_19972_gr}
   \caption{
     Growth rate
   }
   \label{fig:biology-starvation-gr}
  \end{subfigure}

  \begin{subfigure}[htpb]{0.7\textwidth}
   \centering
   \includegraphics[width=\textwidth]{allstrains_19972_budprob}
   \caption{
     Budding probability
   }
   \label{fig:biology-starvation-budprob}
  \end{subfigure}

  \caption{
    Mean growth rate and budding probability of FY4 (blue) and HTB2::mCherry (orange) strains over time during the glucose-starvation experiment, with 95\% confidence intervals shown (shaded).
    Vertical lines (red) show changes in nutrient media.
  }
  \label{fig:biology-starvation-gr-budprob}
\end{figure}



\section{Effect of carbon source on YMC.}
\label{sec:biology-carbon}

\begin{itemize}
\item \emph{Bottom line:} We found that\ldots{}
\begin{itemize}
\item When the cells were grown in a non-fermentable carbon source (20 g/L sodium pyruvate), the cells showed longer metabolic cycles and cell division cycles.  Though the longer cell division cycles were because of longer G1 phases but unchanged S/M phases.  Furthermore, the synchrony between the two oscillators remain, but there were more cases in which the flavin signal peaks without a budding event \textbf{(supplementary figures)}
\begin{itemize}
\item (Add statistical tests to reject the null hypothesis that the mean duration of metabolic cycles in pyruvate is equal to the mean duration of metabolic cycles in high glucose.)
\end{itemize}
\item When the cells were grown in limiting glucose that emulates conditions in a chemostat (10 mg/L glucose), the cells showed longer metabolic cycles and an absence of cell division.  Furthermore, the signals are of low amplitude and low quality, as evidenced by the low signal-to-noise ratios.
\begin{itemize}
\item (Add statistical tests to reject the null hypothesis that the distribution of SNRs are the same between the three nutrient conditions.)
\end{itemize}
\item In contrast to 20 g/L glucose in which metabolic cycles were asynchronous, in these conditions there is some degree of metabolic cycle synchrony between cells. \textbf{(add figures)}
\end{itemize}
\item \emph{Importance of this section:}
\begin{itemize}
\item Confirm \textcite{papagiannakisAutonomousMetabolicOscillations2017} but in flavin, extend in more carbon sources.  It is known that nutrient changes affect the YMC (citation needed).
\item Limiting carbon sources emulate the chemostat.
\end{itemize}
\item \emph{Evidence:}
\begin{itemize}
\item Figures 4-5.
\end{itemize}

\item \textbf{Missing experiments:}
\begin{itemize}
\item Raffinose.  This is a carbon source that \textcite{papagiannakisAutonomousMetabolicOscillations2017} did not use.  It is a trisaccharide and a common non-glucose pre-growth carbon source, certainly in Hxt- and sugar-related studies (Yu).  Yeast must secrete invertase to metabolise it and metabolism on this compound is slow.  So, growth on raffinose emulates a low-glucose concentration and the cell respires.  However, there is a risk that invertase is washed away in microfluidic experiments, so the growth rate should be tested first before subsequent experiments.
\end{itemize}
\end{itemize}

\begin{figure}
  \centering
  \begin{subfigure}[htpb]{1.0\textwidth}
   \centering
   \includegraphics[width=\textwidth]{pyruvate_single_birth_plot_edit.pdf}
   \caption{
     Oscillations of flavin fluorescence lengthen when cells are cultivated in \SI{20}{\gram~\litre^{-1}} pyruvate while the duration of S/M phase stays constant.
     %Figure shows sample flavin fluorescence levels (purple) and histone 2B localisation (pink) in single cells.
     %Vertical lines (black) indicate budding.
   }
   \label{fig:biology-pyruvate-single}

  \end{subfigure}
  \begin{subfigure}[t]{0.4\textwidth}
   \centering
   \includegraphics[width=\textwidth]{htb2mCherry_31594_12.pdf}
   \caption{
     Autocorrelation functions and periods determined by Fourier spectrum of flavin fluorescence and histone 2B levels.
     This figure indicates that the cells' metabolic cycles and cell division cycles are both consistently approximately 4 hours long.
   }
   \label{fig:biology-pyruvate-acf}
  \end{subfigure}%
  \begin{subfigure}[t]{0.4\textwidth}
   \centering
   \includegraphics[width=\textwidth]{pyruvate_xcf_edit.pdf}
   \caption{
    Cross-correlation between flavin and histone 2B signals, indicating that histone levels peak 1 hour after flavin fluorescence peaks.
   }
   \label{fig:biology-pyruvate-xcf}
  \end{subfigure}

  \caption{
    When cultivated in \SI{20}{\gram~\litre^{-1}} pyruvate, cells show longer metabolic cycles while the S/M phase of the cell division cycle remains the same length and the G1 phase lengthens.
  }
  \label{fig:biology-pyruvate}
\end{figure}


\begin{figure}
  \centering
  \begin{subfigure}[htpb]{1.0\textwidth}
   \centering
   \includegraphics[width=\textwidth]{limiting_single_birth_plot_edit.pdf}
   \caption{
     Oscillations of flavin fluorescence are noisier when cells are cultivated in \SI{10}{\milli\gram~\litre^{-1}} glucose while cells show minimal growth or division.
     %Figure shows sample flavin fluorescence (purple) and histone 2B (pink) level in a single cell grown in \SI{10}{\milli\gram~\litre^{-1}}.
     %Vertical dashed lines (black) indicate budding.
   }
   \label{fig:biology-lowglc-single}
  \end{subfigure}

  \begin{subfigure}[htpb]{0.7\textwidth}
   \centering
   \includegraphics[width=\textwidth]{htb2mCherry_31492_12.pdf}
   \caption{
     Autocorrelation functions and periods determined by Fourier spectrum of flavin fluorescence and histone 2B levels.
     This figure indicates that the cells' metabolic cycles are approximately 2 hours long, but the duration of the cell division cycle is not robust nor consistent across the population.
   }
   \label{fig:biology-lowglc-acf}
  \end{subfigure}

  \begin{subfigure}[t]{0.3\textwidth}
   \centering
   \includegraphics[width=\textwidth]{limiting_snr_edit.pdf}
   \caption{
     Signal-to-noise ratios, \SI{10}{\milli\gram~\litre^{-1}} glucose.
     %Distribution of signal-to-noise ratios computed for each flavin time series derived from cells cultivated in \SI{10}{\milli\gram~\litre^{-1}}, showing that most cells have noisy signals and thus implying low-amplitude metabolic cycles.
   }
   \label{fig:biology-lowglc-snr}
  \end{subfigure}%
  \begin{subfigure}[t]{0.3\textwidth}
   \centering
   \includegraphics[width=\textwidth]{pyruvate_snr_edit.pdf}
   \caption{
     Signal-to-noise ratios, \SI{20}{\gram~\litre^{-1}} pyruvate.
    %Distribution of signal-to-noise ratios computed for each flavin time series derived from cells cultivated in \SI{20}{\gram~\litre^{-1}} pyruvate, showing that cells have higher-quality signals.
   }
   \label{fig:biology-pyruvate-snr}
  \end{subfigure}%
  \begin{subfigure}[t]{0.3\textwidth}
   \centering
   \includegraphics[width=\textwidth]{glucose_snr_edit.pdf}
   \caption{
     Signal-to-noise ratios, \SI{20}{\gram~\litre^{-1}} glucose.
    %Distribution of signal-to-noise ratios computed for each flavin time series derived from cells cultivated in \SI{20}{\gram~\litre^{-1}} glucose, showing that cells have higher-quality signals.
   }
   \label{fig:biology-highglc-snr}
  \end{subfigure}

  \caption{
    When cultivated in growth-limiting (\SI{10}{\milli\gram~\litre^{-1}}) glucose, cells show lower-quality metabolic cycles with lower amplitudes, as suggested by the distributions of signal-to-noise ratios computed for each flavin time series (\ref{fig:biology-lowglc-snr},~\ref{fig:biology-pyruvate-snr},~\ref{fig:biology-highglc-snr}).
  }
  \label{fig:biology-lowglc}
\end{figure}


\begin{figure}
  \centering
  \begin{subfigure}[htpb]{0.45\textwidth}
   \centering
   \includegraphics[width=\textwidth]{allstrains_31492_gr}
   \caption{
     Insert caption here
   }
   \label{fig:biology-lowglc-gr}
  \end{subfigure}%
  \begin{subfigure}[htpb]{0.45\textwidth}
   \centering
   \includegraphics[width=\textwidth]{allstrains_26643_gr}
   \caption{
     Insert caption here
   }
   \label{fig:biology-highglc-gr}
  \end{subfigure}

  \begin{subfigure}[htpb]{0.45\textwidth}
   \centering
   \includegraphics[width=\textwidth]{allstrains_31492_budprob}
   \caption{
     Growth rate
   }
   \label{fig:biology-lowglc-budprob}
  \end{subfigure}%
  \begin{subfigure}[htpb]{0.45\textwidth}
   \centering
   \includegraphics[width=\textwidth]{allstrains_26643_budprob}
   \caption{
     Budding probability
   }
   \label{fig:biology-highglc-gr}
  \end{subfigure}

  \caption{
    Growth rates and budding probabilities of cells in limiting glucose (\SI{10}{\milli\gram~\litre^{-1}}) compared to high glucose \SI{20}{\gram~\litre^{-1}}.
    Mean growth rate and budding probability of FY4 (blue) and HTB2::mCherry (orange) strains over time, with 95\% confidence intervals shown (shaded).
  }
  \label{fig:biology-lowglc-gr-budprob}
\end{figure}



\section[Potassium-deficient media]{Do single-cell flavin traces recapitulate dissolved-oxygen YMCs in chemostats? -- potassium-deficient media}
\label{sec:biology-potassium_deficient}
\begin{itemize}
\item \emph{Bottom line:} We found that\ldots{}
\begin{itemize}
\item Single-cell metabolic oscillations are generated in potassium-deficient media, but they are longer and are generated less reliably as in normal minimal media.  This is in contrast to what \textcite{oneillEukaryoticCellBiology2020} suggest: as potassium is gradually replaced with sodium in chemostat culture media, the amplitude of dissolved-oxygen oscillations decrease until they disappear altogether.
\end{itemize}
\item \emph{Importance of this section:}
\begin{itemize}
\item Attempt to show that the single-cell metabolic cycle and the chemostat metabolic cycle are the same cycle.
\item Chemostat experiments obscure the behaviour of individual cells.  My experiments should produce a more detailed picture of how changes to the metabolic cycle affect the coarse-grained readouts in chemostat-based studies.
\end{itemize}
\end{itemize}
\begin{itemize}
\item \emph{Evidence:}
\begin{itemize}
\item Figures ...
\item High-glucose experiment, but with all potassium in the media replaced with sodium.
\end{itemize}

\item \textbf{Missing analysis:}
\begin{itemize}
\item ...
\end{itemize}
\item \textbf{Missing experiments:}
\begin{itemize}
\item ...
\end{itemize}
\end{itemize}

\begin{figure}
  \centering
  \begin{subfigure}[htpb]{1.0\textwidth}
   \centering
   \includegraphics[width=\textwidth]{htb2mCherry_613_plots_single_htb2mCherry012_90_2_adapted.pdf}
   \caption{
     Flavin fluorescence and cell division cycles are maintained before, during, and after perfusion of potassium-deficient nutrient media (FY4 HTB2::mCherry).
   }
   \label{fig:biology-kdeficient-single}
  \end{subfigure}

  \begin{subfigure}[htpb]{0.7\textwidth}
   \centering
   \includegraphics[width=\textwidth]{htb2mCherry_613_12.pdf}
   \caption{
     Autocorrelation functions and periods determined by Fourier spectrum of flavin fluorescence and histone 2B levels.
     This figure indicates that the cells' metabolic cycles and cell division cycles are both consistently 120 minutes long.
   }
   \label{fig:biology-kdeficient-acf}
  \end{subfigure}

  \caption{
    When grown in potassium-deficient media and with \SI{20}{\gram~\litre^{-1}} glucose as a carbon source, cells showed metabolic cycles, contrary to \textcite{oneillEukaryoticCellBiology2020}
  }
  \label{fig:biology-kdeficient}
\end{figure}


\begin{figure}
  \centering
  \begin{subfigure}[htpb]{0.9\textwidth}
   \centering
   \includegraphics[width=\textwidth]{613_distribs_0300}
   \caption{
     \SI{300}{\minute}
   }
   \label{fig:biology-kdeficient-distribs-0300}
  \end{subfigure}

  \begin{subfigure}[htpb]{0.9\textwidth}
   \centering
   \includegraphics[width=\textwidth]{613_distribs_0360}
   \caption{
     \SI{360}{\minute}
   }
   \label{fig:biology-kdeficient-distribs-0360}
  \end{subfigure}

  \begin{subfigure}[htpb]{0.9\textwidth}
   \centering
   \includegraphics[width=\textwidth]{613_distribs_0420}
   \caption{
     \SI{420}{\minute}
   }
   \label{fig:biology-kdeficient-distribs-0420}
  \end{subfigure}

  \begin{subfigure}[htpb]{0.9\textwidth}
   \centering
   \includegraphics[width=\textwidth]{613_distribs_0480}
   \caption{
     \SI{480}{\minute}
   }
   \label{fig:biology-kdeficient-distribs-0480}
  \end{subfigure}

  \begin{subfigure}[htpb]{0.9\textwidth}
   \centering
   \includegraphics[width=\textwidth]{613_distribs_0700}
   \caption{
     \SI{700}{\minute}
   }
   \label{fig:biology-kdeficient-distribs-0700}
  \end{subfigure}

  \begin{subfigure}[htpb]{0.9\textwidth}
   \centering
   \includegraphics[width=\textwidth]{613_distribs_1080}
   \caption{
     \SI{1080}{\minute}
   }
   \label{fig:biology-kdeficient-distribs-1080}
  \end{subfigure}

  \caption{
    Distribution of quantities at selected time points in the potassium-deficiency experiment, potassium-deficiency being \SIrange{360}{960}{\minute}.
    HTB2::mCherry and flavin signals were based on raw data extracted from the fluorescent images.
    Growth rate was calculated based on the rolling mean of the time point-to-time point change in cell volume estimated by \emph{aliby}, and the budding probability was calculated based on the rolling mean of the boolean vector that indicates the presence or absence of budding event over time.
  }
  \label{fig:biology-kdeficient-distribs}
\end{figure}


\begin{figure}
  \centering
  \includegraphics[width=\textwidth]{613_ks_highlight}
  \caption{
    Potassium-deficiency experiment.
    For each quantity, the Kolmogorov-Smirnov test statistics from two-tailed tests between the distribution at \SI{300}{\minute} and the distribution at each time point of interest is shown.
    The null hypothesis for each test is that the two samples are drawn from the same probability distribution, and thick lines indicate where the hypothesis is rejected ($p < 0.05$).
    Vertical lines (red) indicate changes in nutrient conditions.
  }
  \label{fig:biology-kdeficient-ks}
\end{figure}


\begin{figure}
  \centering
  \begin{subfigure}[htpb]{0.7\textwidth}
   \centering
   \includegraphics[width=\textwidth]{allstrains_613_gr}
   \caption{
     Growth rate
   }
   \label{fig:biology-kdeficient-gr}
  \end{subfigure}

  \begin{subfigure}[htpb]{0.7\textwidth}
   \centering
   \includegraphics[width=\textwidth]{allstrains_613_budprob}
   \caption{
     Budding probability
   }
   \label{fig:biology-kdeficient-budprob}
  \end{subfigure}

  \caption{
    Mean growth rate and budding probability of FY4 (blue) and HTB2::mCherry (orange) strains over time during the potassium-deficiency experiment, with 95\% confidence intervals shown (shaded).
    Vertical lines (red) show changes in nutrient media.
  }
  \label{fig:biology-kdeficient-gr-budprob}
\end{figure}



\section[Deletion strains]{Do single-cell flavin traces recapitulate dissolved-oxygen YMCs in chemostats? -- deletion strains}
\label{sec:biology-deletions}

\begin{itemize}
\item \emph{Bottom line:} We found that\ldots{}
\begin{itemize}
\item Single-cell metabolic oscillations are not reliably generated from the auxotrophic zwf1$\Delta$ strain, which has been shown to not generate dissolved-oxygen oscillations in the chemostat (\textcite{tuCyclicChangesMetabolic2007}).
\item Single-cell metabolic oscillations are also not reliably generated from the auxotrophic tsa1$\Delta$ tsa2$\Delta$ strain, which has been generate dissolved-oxygen chemostat oscillations of a different waveform as the wild type strain (\textcite{caustonMetabolicCyclesYeast2015}).  I also observe cells getting sick, i.e.\ bloating and clogging earlier than the wild type.
\item Single-cell metabolic oscillations are generated by certain prototrophic deletion strains (swe1$\Delta$, rim11$\Delta$, and tsa1$\Delta$ tsa2$\Delta$) and have properties that differ from wild-type.  However, these differences do not reflect how dissolved-oxygen oscillations differ from their wild-type equivalents (\textcite{caustonMetabolicCyclesYeast2015}).
\end{itemize}
\item \emph{Importance of this section:}
\begin{itemize}
\item Attempt to show that the single-cell metabolic cycle and the chemostat metabolic cycle are the same cycle.
\item Chemostat experiments obscure the behaviour of individual cells.  My experiments should produce a more detailed picture of how changes to the metabolic cycle affect the coarse-grained readouts in chemostat-based studies.
\item With deletion strains, I can give some mechanistic insight on YMC, otherwise largely unknown.
\begin{itemize}
\item \textcite{tuCyclicChangesMetabolic2007}
\begin{itemize}
\item zwf1$\Delta$: metabolic cycles abolished, key metabolic enzyme that catalyses production of NADPH, which is a key player in YMC (though other enzymes can compensate NADPH production).
\end{itemize}
\item \textcite{caustonMetabolicCyclesYeast2015}
\begin{itemize}
\item swe1$\Delta$: gene responsible for CDC processes (another biological rhythm) e.g. DNA repair.  Deletion shown to affect CDC-YMC coupling.
\item rim11$\Delta$: gene involved in circadian rhythm (another biological rhythm).  Deletion strain shown to have shorter YMCs.
\item tsa1$\Delta$ tsa2$\Delta$: gene involved in redox metabolism (key player of metabolic cycle), linked to circadian rhythm (another biological rhythm).  Deletion strain shown to have shorter YMCs and of a different waveform, i.e. an additional `dip' in dissolved oxygen corresponding to the reductive-charging phase.
\end{itemize}
\end{itemize}
\end{itemize}
\item \emph{Evidence:}
\begin{itemize}
\item Figures 6-10.
\item We include high-glucose to be consistent with my previous section.  Also including limiting glucose as it approximates chemostat conditions (\textcite{jonesCyberneticModelGrowth1999}, and add other citations that say that chemostat is starvation).  If chemostat conditions mean starvation, I should look at microfluidic conditions that mimic that best.
\item Precultured in pyruvate because zwf1$\Delta$ doesn't grow well on high glucose liquid culture (add evidence in the form of growth rate graphs), and tsa1$\Delta$ tsa2$\Delta$ follows for consistency.  Different amino acid formulation for BY4742 vs BY4741 because different auxotrophy.
\item The properties of the single-cell metabolic cycles we observe in Causton's strains differ from the properties of chemostat cycles \textcite{caustonMetabolicCyclesYeast2015} describe.  Under starvation, the super-long CDCs suggested by chemostat studies are possible (citations needed).  However, the period of the metabolic cycles I observe are much shorter than chemostat-based metabolic cycles.  Furthermore, the shape of the metabolic cycles is different -- notably tsa1$\Delta$ tsa2$\Delta$ that doesn't show a `notch' as \textcite{caustonMetabolicCyclesYeast2015} showed.
\end{itemize}

\item \textbf{Missing analysis:}
\begin{itemize}
\item Sub-populations, i.e. are there sub-populations that behave differently?  It seems like at least there is a group of cells that produce just noise and a group of cells that produce long-period oscillations, and these oscillations occur over a larger spread of periods than the wild-type -- I have done some analysis that could show the latter.  This model looks likely for zwf$\Delta$ judging by the signal-to-noise ratio histogram.  This is where time series featurisation and clustering (hierarchical, UMAP, etc.) may be useful.  But are there methods that don't rely on machine learning -- consult notes with Andrew Millar.
\item A supplementary figure should try to show equivalence between FY4 and CEN.PK (and BY4741).  There should be an experiment in which I put FY4 and CEN.PK in the same conditions. \textcite{baumgartnerFlavinbasedMetabolicCycles2018} state that the two strains have equivalent metabolic cycles, so there is some precedent, but they didn't do a direct comparison.
\end{itemize}
\item \textbf{Missing experiments:}
\begin{itemize}
\item zwf1$\Delta$ in low glucose.  Problem: auxotrophic.  Need to engineer a deletion in CEN.PK, likely with CRISPR.
\item Chemostat/turbidostat.  But we lack the expertise, and this is its own avenue entirely.  Chemostat vs single-cell equivalence can be its own PhD project.
\end{itemize}
\end{itemize}

\begin{figure}
  \centering
  \begin{subfigure}[t]{0.45\textwidth}
   \centering
   \includegraphics[width=\textwidth]{zwf1egf_409_6.pdf}
   \caption{
     Median flavin fluorescence signal across cells,%; each data point is the median of flavin fluorescence at each time point relative to the first birth.
     suggesting that flavin oscillations have some degree of synchrony with the cell division cycle, and oscillate at a period of approximately \SI{3}{\hour}.
   }
   \label{fig:biology-zwf1-median}
  \end{subfigure}%
  \begin{subfigure}[t]{0.45\textwidth}
   \centering
   \includegraphics[width=\textwidth]{zwf1egf_409_12.pdf}
   \caption{
    Autocorrelation function of flavin fluorescence across cells, suggesting an average oscillation of approximately \SI{3}{\hour}, but with low robustness.
   }
   \label{fig:biology-zwf1-median}
  \end{subfigure}

  \begin{subfigure}[t]{0.45\textwidth}
   \centering
   \includegraphics[width=\textwidth]{zwf1egf_409_10.pdf}
   \caption{
    Distribution of signal-to-noise ratios computed for each flavin time series derived from zwf1$\Delta$ cells, showing a wide distribution of oscillation quality.
   }
   \label{fig:biology-zwf1-snr}
  \end{subfigure}%

  \caption{
    zwf1$\Delta$ (BY4741) cells in \SI{10}{\gram~\litre^{-1}} glucose generate metabolic oscillations.
  }
  \label{fig:biology-zwf1}
\end{figure}


\begin{figure}
  \centering
  \begin{subfigure}[t]{0.45\textwidth}
   \centering
   \includegraphics[width=\textwidth]{tsa1tsa2morgan_1649_6.pdf}
   \caption{
     Median flavin fluorescence signal across cells,%; each data point is the median of flavin fluorescence at each time point relative to the first birth.
     suggesting that flavin oscillations has weak synchrony with the cell division cycle, and oscillate at a period of approximately \SI{3}{\hour}.
   }
   \label{fig:biology-tsa1tsa2-median}
  \end{subfigure}%
  \begin{subfigure}[t]{0.45\textwidth}
   \centering
   \includegraphics[width=\textwidth]{tsa1tsa2morgan_1649_12.pdf}
   \caption{
    Autocorrelation function of flavin fluorescence across cells, showing a lack of oscillations at a regular frequency across the population, in contrast to more regular oscillations in the wild type.
   }
   \label{fig:biology-tsa1tsa2-median}
  \end{subfigure}

  \begin{subfigure}[t]{0.45\textwidth}
   \centering
   \includegraphics[width=\textwidth]{tsa1tsa2morgan_1649_10.pdf}
   \caption{
    Distribution of signal-to-noise ratios computed for each flavin time series derived from tsa1$\Delta$ tsa2$\Delta$ cells, showing that the quality of oscillations is comparable to growth in pyruvate.
   }
   \label{fig:biology-tsa1tsa2-snr}
  \end{subfigure}%

  \caption{
    tsa1$\Delta$ tsa2$\Delta$ (BY4742) cells in \SI{10}{\gram~\litre^{-1}} glucose do not robustly generate metabolic oscillations.
  }
  \label{fig:biology-tsa1tsa2}
\end{figure}


\section{Discussion}
\label{sec:biology-discussion}

% [MOVED FROM BEGINNING]
In this study, I show that yeast cells exhibit oscillations in the fluorescence of flavins that are consistent with previous descriptions of the yeast metabolic cycle.
These oscillations synchronise with the cell division cycle in permissive conditions.
Such flavin oscillations showed different behaviour in respiratory conditions.
Additionally, I found that cells were able to autonomously reset the phase of their metabolic cycle, independently of the cell division cycle, in response to abrupt nutrient changes.
Finally, I showed that deletion strains generated flavin oscillations that exhibited different behaviour from dissolved oxygen oscillations from chemostat conditions.

My results confirm that the metabolic cycle is an autonomous oscillator, independent of the cell division cycle, and responds to perturbations in the environment\ldots{}

% [COLLECTION OF MISSING EXPERIMENTS/CAVEATS/ETC]
\textbf{Permissive conditions missing experiments:}
\begin{itemize}
\item Record NAD(P)H oscillations.  If possible, do it with flavin in the same cells.  If not, then compare these oscillations with the timing of birth events.  This experiment will strengthen the evidence that I am observing single-cell YMCs.  \textcite{baumgartnerFlavinbasedMetabolicCycles2018} didn't do this; maybe that's why they haven't had many citations.
\item fas1$\Delta$ to verify the role of lipids and attempt to rescue with lipid sources (e.g. glycerol trihexanoate, glycerol trioctanoate). \emph{Will start another avenue of exploration entirely, i.e. what makes up the flavin signal in a cell}
\end{itemize}

\subsection{Interpretation of results}
\label{subsec:biology-discussion-interpretation}
\begin{enumerate}
\item Presence of flavin-based single-cell metabolic cycles, which are autonomous and gate the cell division cycle.
\begin{itemize}
\item Results confirm \textcite{papagiannakisAutonomousMetabolicOscillations2017} and \textcite{baumgartnerFlavinbasedMetabolicCycles2018}.  In high glucose, flavin cycles are asynchronous between cells and peaks in flavin fluorescence coincide with bud formation.
\begin{itemize}
\item We expect the flavin fluorescence to peak when buds form, and this is our reasoning:
\begin{itemize}
\item NAD(P)H fluorescence peaks -- i.e. the molecule is in the reduced NAD(P)H form -- when buds form (\textcite{papagiannakisAutonomousMetabolicOscillations2017}).  Flavins should be in the oxidised form when NAD(P)H is in the reduced form because the flavoprotein lipoamide dehydrogenase is in redox equilibrium with NAD(P)H (\textcite{sianoNADHFlavinFluorescence1989}, and also confirmed by several reactions catalysed by \href{Flavoproteins.org}{Flavoproteins} (example proteins and citations needed).  Flavins fluoresce in the oxidised form, thus we expect flavin fluorescence to peak. \textcite{murrayRedoxRegulationRespiring2011} confirms that flavin and NAD(P)H fluorescence to peak in phase in continuous chemostat cultures.
\end{itemize}

\begin{itemize}
\item It would be nice to support this logic with the roles of flavoproteins (\textcite{gudipatiFlavoproteomeYeastSaccharomyces2014}).  However, it is difficult because we can't readily assume the behaviour of individual flavoproteins based on the behaviour of their aggregrate.
\end{itemize}
\end{itemize}
\item Results in pyruvate reveal that as the metabolic cycle lengthens, G1 lengthens but S/M stays the same length, reinforcing a model in which a specific phase of the metabolic cycle gates entry into the cell division cycle.  This is consistent with \textcite{oneillEukaryoticCellBiology2020} and (add other citations here).
\item Cross-correlation between flavin and mCherry signals further confirm the sequence of events while the two biological oscillations (metabolic cycle \& cell division cycle) progress.  This is more obvious in pyruvate.  Specifically, when the new bud forms, the mother cell's flavin fluorescence peaks.  Subsequently, the cell synthesises new histones in S phase, then the cell enters anaphase -- characterised by the sharp drop in the mCherry signal -- as the flavin fluorescence is at its trough.
\item Metabolic cycles still occur even when cells do not divide.  This holds true for `one-off' skipping of cell division and conditions in which cells pause cell division for long periods of time.  My observations add to \textcite{baumgartnerFlavinbasedMetabolicCycles2018} that stopped cell divisions using alpha-factor; I stopped them in a different way.
\item Confirms \textcite{ozsezenInferenceHighLevelInteraction2019}.
\end{itemize}
\item Cells individually generate metabolic cycles and individually adapt them in response to nutrient changes.
\begin{itemize}
\item Literature disputes whether metabolic oscillations arise from interactions between cells or whether cells individually generate oscillations.
\item Even though the cells are physically separated in traps and nutrient media flows through them, single-cell flavin oscillations can synchronise and reset phase in certain conditions.  Diffusible metabolites cannot be responsible.  We thus conclude that each cell -- on its own -- can reset its metabolic cycle in response to nutrient changes.
\begin{itemize}
%\item Discussion: back-of-the-envelope physics calculations to show that diffusion at the timescale relevant to the experiments cannot be responsible for cell-to-cell synchrony given the physical distance between the cells and the flow rate.
\item Biochemical interpretation for the cell division cycle: if starvation occurs before START, cell remain in pause; if starvation occurs after START, cells go into pause.  Cells stuck in M phase during starvation?
\item Biochemical interpretation for the metabolic cycle: What biochemical mechanism does the cell use to reset the phase of its metabolic cycle?  How does this fit in with my model of explaining how the cycle works and coordinates cellular events (metabolism \& nutrient mobilisation) with the environment?  Link to FBA?
\end{itemize}
\item Doesn't necessarily mean that cells \emph{cannot} communicate with each other, but communication is \emph{not required}.
\end{itemize}
\item Cells adapt their metabolic cycle to nutrient conditions (\textcite{papagiannakisAutonomousMetabolicOscillations2017}, other citations may be useful)
\begin{itemize}
\item The behaviour of the metabolic cycle changes when the cells are grown on low glucose or on pyruvate.
\begin{itemize}
\item Biochemical interpretation: nutrient conditions that favour respiration over fermentation -- and thus slower growth rate -- leads to slower YMCs, but why?
\item Do the periods conform to those reported by \textcite{papagiannakisAutonomousMetabolicOscillations2017}?  If not, why?
\end{itemize}
\end{itemize}
\item Reconciling chemostat and single-cell studies.

\begin{itemize}
\item In chemostats, glucose is limiting (\textcite{jonesCyberneticModelGrowth1999}).  Synchrony of metabolic cycles between cells in respiring conditions may explain observations in chemostat as cells are near glucose starvation in these conditions.
\item Single-cell metabolic cycles in zwf1$\Delta$ are inconsistent, and this may explain the absence of dissolved-oxygen oscillations in the chemostat.  zwf1$\Delta$ affects many metabolic processes; most notably, it removes a major pathway of NAD(P)H generation from reduction of NAD(P)+.  The most abundant flavoproteins involve NAD(P)H redox, so it it reasonable to believe that flavin oscillations are affected.  Perhaps zwf1$\Delta$ impairs the metabolic cycle in some way.
\item A similar logic likely applies to tsa1$\Delta$ tsa2$\Delta$ -- perhaps the interaction between a more variable length of metabolic oscillations interacts with population synchrony, or lack thereof, in the chemostat.  Furthermore, the tsa1$\Delta$ tsa2$\Delta$ double deletion affects the peroxiredoxin-thioredoxin system, therefore affecting redox balance, likely in an extensive way.
\item The fact that metabolic oscillations are present when the medium is potassium-free can be seen as more evidence that there is no straight one-to-one correspondence between dissolved-oxygen metabolic oscillations in the chemostat and single-cell flavin oscillations from single cells.
\item Therefore, the difference between single-cell and chemostat traces warrant a model to explain the differences.  Likely something to do with population.  Highlights weakness of chemostat experiments.
\begin{itemize}
\item See e.g. \textcite{burnettiCellCycleStart2016}
\end{itemize}
\end{itemize}
\item Our analysis methods add on to those used in previous single-cell studies of the metabolic cycle.
\begin{itemize}
\item This is important because data processing steps can substantially alter the conclusions of any time series analysis study.
\item We pre-process the time series better.  Specifically, we use a Butterworth filter rather than sliding-window or Savitzky-Golay detrending so we have better control over the frequencies.
\item We use several analysis methods, some from the circadian rhythm literature, in conjunction with each other to confirm features of the oscillatory signals.  We use cross-correlation and Granger causality analysis to quantify the sequence of events in biological oscillators and these methods utilise more information than comparing peaks of time series as performed previously.
\end{itemize}
\end{enumerate}

\subsection{Broader implications}
\label{sec:biology-discussion-implications}

\begin{itemize}
\item Confirms utility of using flavin as an indicator of YMC.  Confirms utility of our microfluidics platform.  Our technology produces vast amounts of data, so our computational tools are especially useful.  Confirms utility of our analysis methods.
\item Confirms model in which autonomous, single-cell YMC responds to nutrient conditions and adjusts period/phase accordingly.  YMC-CDC synchrony holds in a window of frequencies -- outside this, YMCs still occur but coupling changes.  Points to model that YMCs govern cell resources and only if they are adequate does the cell initiate a CDC.  Suggestion that deletion strain impair this model.
\item Autonomy of YMCs (cell-autonomous and independence from CDC), starvation in chemostat, and presence of subpopulations may explain differences between chemostat and single-cell based studies.  My study makes the case for studying this system in single-cell as it gives more info than chemostat.
\item Something fundamental to control of biological rhythms?  Links to other processes?
\begin{itemize}
\item pH and growth rate oscillations (\textcite{luziaPHDependenciesGlycolytic})
\end{itemize}
\end{itemize}
