% TODO:
% - Editing
%   - Add/Read citations where marked
%   - Chemostat vs single-cell writing
%   - Reorganise end of flavins in YMC section
%   - Internal links/refs
% - Add figures & tables

\chapter{Introduction}

\section{Motivation of thesis}

This thesis aims to understand how an organism adapts its metabolism and cellular processes in response to external conditions.
I do so by through using the yeast metabolic cycle (YMC) as a framework for biological oscillators.
My reasons are twofold: (a) biological oscillators are important for coordination of responses and are present across kingdoms, (b) there are unanswered questions about the mechanistic basis of the YMC and about reconciling evidence from two types of experimental studies.
Therefore, I study YMC regulation in isolated cells in different nutrient conditions.

% - describe the logic of the rest of the chapters
% TODO: replace chapter numbers with \ref{}
This thesis is divided into six chapters:
\begin{enumerate}
  \item Chapter 1 discusses the background behind the yeast metabolic cycle and using flavin autofluorescence as a way to monitor the yeast metabolic cycle.
  \item Chapter 2 discusses the methods: single-cell microfluidics of yeast cells followed by an automated image analysis pipeline.
  \item Chapter 3 discusses the analysis of oscillatory time series; given the size of the datasets and the challenges of analysing noisy low-resolution time series, this deserves discussion in its own right.
  This chapter will step through the process of analysis and provide a review \& justification of the computational methods at each stage.
  \item Chapter 4 presents biological results, using the analysis methods discussed in chapter 3.
  I show that the metabolic cycle and cell division cycle are autonomous and synchronise in permissive conditions, while perturbations affect the relationship between these two biological oscillators.
  \item Chapter 5 discusses using flux balance analysis to answer whether temporal partitioning of biosynthesis under proteome constraints explains the timing of the yeast metabolic cycle.
  \item Finally, chapter 6 combines previous understanding of the metabolic cycle, experimental observations, and mathematical models to propose a coarse-grained, phenomenological model of the yeast metabolic cycle.
  This chapter also suggests further avenues of study.
\end{enumerate}


\section{Yeast metabolic cycle}
\label{sec:intro-ymc}

\subsection{Introduction to biological rhythms}
\label{subsec:intro-ymc-biological_rhythms}

\subsubsection{Biological basis of biological rhythms}
\label{subsubsec:intro-ymc-biological_rhythms-biological_basis}
% Physiological importance of biological rhythms
% including the circadian rhythm, cell division cycle, yeast metabolic cycle
% Basis, e.g. biochemical oscillators

% Literature:
% Mellor (2016) -- provides a good overview.
% and that collection of review articles probably give good definitions

[FIGURE: SHOW COMPONENTS OF A BIOLOGICAL OSCILLATOR, ADAPTED FROM MELLOR 2016]

Biological rhythms are repeating physiological or cellular processes.
Genetic oscillators, biochemical oscillators, and metabolic oscillators, all linked to a cellular redox cycle, govern biological rhythms \citep{mellorMolecularBasisMetabolic2016}.
Biological rhythms can occur at different time scales, from seconds (e.g.\ glycolytic cycle), to ultradian cycle (i.e.\ cycles that are more frequent than 24 hours), to circadian rhythms (24 hours).
Biological rhythms are important in temporally separating physiological processes.
This is instrumental in responding to external conditions, including nutrient conditions, growth requirements, or the day-night cycle.
Thus, this means that biological rhythms can vary according to conditions.

[FIGURE(S) MAY BE USEFUL IN DEMONSTRATING THE CELL DIVISION CYCLE HERE.  SHOW G1/S/G2/M PHASES, SHOW OVERVIEW OF CDKs, ETC.]

% VERY useful citation: adlerYeastCellCycle2022

Such biological rhythms include the circadian rhythm and the cell division cycle.
To demonstrate the definition of biological rhythm, I will discuss here the cell division cycle, which is well-characterised.
The cell division cycle in budding yeast is governed by a series of gene regulatory networks that interact in a feedback loop, resulting in oscillatory expression of regulatory proteins, namely, cyclin-CDK complexes that regulate cellular events in a temporal manner \citep{adlerYeastCellCycle2022, orlandoGlobalControlCellcycle2008, murrayRecyclingCellCycle2004}.
This cycle also includes biochemical and metabolic oscillators, for example, biosynthesis during S phase.
As with other biological rhythms, the cell division cycle also includes a system to control it, so that DNA replication occurs once every cell division cycle and so that the cell only divides when necessary.
The importance of such control systems are highlighted by disorders when these systems are impaired, such as chromosome aberrations and cancer.

The yeast metabolic cycle is a type of biological rhythm because it has the properties that define a biological rhythm.
Namely, it has gene-expression oscillators as evidenced by transcript cycling in its phases,
it has biochemical oscillators as evidenced by changes in dissolved oxygen in the chemostat,
and it has metabolite oscillations as evidence by changes in the levels of compounds that undergo redox reactions like NADH/NADPH and flavins.
I discuss further this evidence in the context of the known progression of the yeast metabolic cycle in section \ref{subsubsec:intro-ymc-definition-phases}.
However, in contrast to the cell division cycle, the control mechanisms of the yeast metabolic cycle are less well characterised --- I discuss this further in section \ref{subsec:intro-ymc-unresolved}.

\subsubsection{Theoretical basis of biological rhythms}
\label{subsubsec:intro-ymc-biological_rhythms-theoretical_basis}
% - Mathematics of systems of coupled oscillators.

% Single oscillations
% Useful here: cell division cycle modelling literature, e.g. Novak/Tyson,
% Adler et al. (2022) -- comprehensive review of cell division cycle models
% Goldbeter (2022) -- models of several examples
The theoretical basis of biological rhythms originated in work in the 1960s, which included simple systems of ordinary differential equations to describe negative feedback control circuits \citep{goodwinOscillatoryBehaviorEnzymatic1965, griffithMathematicsCellularControl1968}.
Experimental observations have then informed the development of models with finer detail.
Furthermore, synthetic genetic circuits have also been modelled and developed \citep{elowitzSyntheticOscillatoryNetwork2000}.

To illustrate a natural biological rhythm, I discuss the cell division cycle.
The well-characterised cell division cycle has inspired models with a variety of approaches.
Early models are based on a negative feedback loop of key components as identified by experimental studies.
For example, \textcite{goldbeterMinimalCascadeModel1991} assumed a minimal model of one cyclin, one kinase, and one protease to construct a negative feedback loop with a delay, giving rise to stable oscillations.
Such a strategy forms the basis of later models that incorporate more detail, including additional control points of the cell division cycle \parencite{chenIntegrativeAnalysisCell2004}, responses to perturbations such as osmotic stress \parencite{adroverTimeDependentQuantitativeMulticomponent2011}, and relationship with other oscillators like the circadian rhythm \parencite{gerardEntrainmentMammalianCell2012, charvinForcedPeriodicExpression2009, droinLowdimensionalDynamicsTwo2019}.
More recent, comprehensive models include \textcite{adlerYeastCellCycle2022} which is based on a system of ordinary differential equations adapted for the modelling to pheromone and osmotic shock responses, and \textcite{novakMitoticKinaseOscillation2022}, which models the cell division cycle as a series of switches between two stable steady states whose behaviour is regulated by the CDK oscillator.

In contrast, less well-characterised natural biological rhythms have given rise to models with fewer detail and precision.
An example is the glycolytic oscillation.
The glycolytic oscillation is a type of ultradian biochemical oscillator, characterised by oscillations in NADH levels (and other cofactors) in budding yeast cells at the time scale of 10-ish seconds \parencite{doddLiveCellImaging2017, lloydSaccharomycesCerevisiaeOscillatory2019, olsenOscillationsYeastGlycolysis2021}, when they are in high-glucose conditions.
A recent model is a coarse-grained model
driven by positive feedback loops as a system of two coupled instability-generating mechanisms \parencite{goldbeterMultisynchronizationOtherPatterns}.

% Forced oscillators, coupled oscillators
% Useful here: \textcite{tysonTimekeepingDecisionmakingLiving} and related reviews
As biological rhythms are often coupled with each other, forced and coupled oscillators have been modelled.
If an oscillator is forced, it has a natural oscillation frequency, but is forced from it due to an external force applied at a regular interval.
An example is the circadian clock, which is entrained to the light-dark cycle \parencite{goldbeterMultisynchronizationOtherPatterns}.
Yeast glycolytic oscillations can also be entrained via a periodic input of substrate.
Forced oscillators are closely linked to coupled oscillators, in which two oscillators are coupled to each other by certain activation/deactivation events.
Two coupled oscillators tend to oscillate at a compromise frequency if the natural frequencies of each are close enough to each other.
Otherwise, complex oscillations can occur: the oscillators lock to a rational ratio of frequencies --- i.e.\ one oscillator goes through $p$ periods while the other goes through $q$ periods.
In this case, the exact ratio depends on the ratio of the natural frequencies.
Furthermore, in certain cases, chaos can occur.
There is a mathematical basis in Arnold tongues \citep{heltbergTaleTwoRhythms2021}.
Experimental observations support this.
For example, \citet{charvinForcedPeriodicExpression2009} showed that externally forcing cell division cycles via glucose pulsing leads to phase-locking of the cell division cycle oscillator only within a range of extrinsic periods.

The yeast metabolic cycle has been modelled as a system of coupled oscillators \citep{papagiannakisAutonomousMetabolicOscillations2017,ozsezenInferenceHighLevelInteraction2019}, based on how it is linked to the cell division cycle.
I will discuss this in section \ref{subsec:intro-ymc-model}.

\subsection{Definition and description of the yeast metabolic cycle}
\label{subsec:intro-ymc-definition}
% Worth re-reading: Mellor 2016, Lloyd 2019

% Not too sure if the subsubsections in this subsection are a good idea.
\subsubsection{History of evidence for the yeast metabolic cycle}
\label{subsubsec:intro-ymc-definition-history}

Aspects of the YMC have been observed over decades.
\citet{nosohSYNCHRONIZATIONBUDDINGCYCLE1962} discovered that synchronised \emph{S. cerevisiae} cultures show oscillatory oxygen consumption.
\citet{kasparvonmeyenburgEnergeticsBuddingCycle1969} showed that gas metabolism and energy generation increase upon budding, while \citet{mochanRespiratoryOscillationsAdapting1973} described a high-amplitude respiratory oscillation following a substrate shift from glucose to ethanol.
\citet{satroutdinovOscillatoryMetabolismSaccharomyces1992} were the first to describe the metabolic components of a 40-minute YMC for cells in continuous culture.
\citet{tuLogicYeastMetabolic2005}
first incorporated transcript cycling in the description of the YMC and defined the YMC events,
based on a chemostat-based investigation of growth of budding yeast on glucose-starved conditions.

The yeast metabolic cycle is longer in duration and is more robust than a similar biological oscillator, the glycolytic oscillation.
The glycolytic oscillation has a period on the scale of 40 seconds \citep{olsenRegulationGlycolyticOscillations2009}.
In contrast, the yeast metabolic cycle has been described, using various definitions, to either exhibit a 40-minute short-phase cycle \citep{lloydUltradianMetronomeTimekeeper2005, liRapidGenomescaleResponse2006, lloydRedoxRhythmicityClocks2007}, or a long-phase cycle, which is most commonly described to be 4-5 hours \citep{tuLogicYeastMetabolic2005, tuCyclicChangesMetabolic2007}, but also ranges between 1.4 hours to 14 hours, depending on the chemostat dilution rate \citep{beuseEffectDilutionRate1998, oneillEukaryoticCellBiology2020}.

Glycolytic oscillations are highly damped, but yeast metabolic oscillations are robust, lasting for weeks \parencite{lloydRedoxRhythmicityClocks2007}.
Additionally, glycolytic oscillations have been observed in anaerobic conditions \citep{lloydSaccharomycesCerevisiaeOscillatory2019}, but yeast metabolic cycles have been observed in aerobic conditions.
Moreover, glycolytic oscillations are characterised by fluctuations in NADH fluorescence only, but yeast metabolic cycles consist of fluctuations in NADH fluorescence as well as fluctuations in other compounds like ATP and flavins.

\subsubsection{Phases of the yeast metabolic cycle}
\label{subsubsec:intro-ymc-definition-phases}
% INTEGRATE THIS STRUCTURE IN EACH PARAGRAPH THAT DISCUSSES EACH PHASE
% 1. Overarching theme of phase
% 2. Cellular events (cell division cycle, mitochondria)
% 3. Metabolic events (metabolite concentrations, redox)
% 4. Transcript/genetic events.
% Or any that make sense, e.g. 3 main events and all the evidence from different
% parts of biochemistry.
% Structure should then link well with the definition of biological rhythms in
% a previous subsection.
% ---------------------------------------------------------------------
Based on chemostat studies,
the YMC can be divided into two major phases: an oxidative, high-oxygen consumption (OX/HOC) phase and a reductive, low-oxygen consumption (RED/LOC) phase.
% Though Lloyd/Murray camp also describe these 3 phases
Many authors \citep{slavovMetabolicCyclingCell2011, murrayRedoxRegulationRespiring2011, caustonMetabolicRhythmsFramework2018} use oxygen consumption rates, evidenced by the change of dissolved oxygen concentrations over time, as a basis to refer to the YMC as a two-phase cycle.
Though, there are authors \citep{machneYinYangYeast2012} that base their two-phase model on the clustering of gene expression patterns.
\citet{krishnaMinimalPushPull2018} interpret the oxidative phase as a growth state, while the reductive phase is a quiescent state.
In contrast to the two-phase model, some authors identify a three-phase model with a reductive-building (RB) phase and a reductive-charging (RC) phase within the reductive phase, especially within the long-phase (4--5 hours) yeast metabolic cycle.
This three-phase model is primarily based on cellular events, including clustering of transcript trajectories \citep{tuLogicYeastMetabolic2005} and of metabolite concentration trajectories \citep{tuCyclicChangesMetabolic2007}.

Single-cell studies \citep{papagiannakisAutonomousMetabolicOscillations2017, baumgartnerFlavinbasedMetabolicCycles2018} do not discuss phases as the single-cell microfluidic set-up does not allow live monitoring of transcription, and oxygen consumption rate is an emergent property from chemostat cultures.
% ---- I don't think any publication explicitly says this -- it's mostly just Kevin IIRC.  I suggest phrasing it differently, below
%However, others [INSERT CHAIN OF PUBLICATIONS HERE] debate the existence of the reductive-charging phase and argue that this phase is an artefact of cellular adaptation to glucose limitation or feast-and-famine conditions.
Possibly, the two- or three-phase response results from cellular adaptation to glucose limitation in chemostat cultures, and it is unknown whether these dynamics hold true in glucose-rich conditions, which cannot be created in a chemostat \citep{slavovCouplingGrowthRate2011}.

[ADD DIAGRAM HERE -- MELLOR HAS A GOOD ONE, ADAPT HERS]

In the oxidative phase, cells consume oxygen at a high rate as respiration, fermentation, and
% Potentially may be the reason auxotrophs (hypothetically) don't have YMCs --
% but I showed that they do
energy-demanding processes
like biosynthesis and gene expression occur.
Occurrence of biosynthesis and associated gene expression is confirmed by increased transcripts from genes encoding components of the translation machinery and amino acid biosynthesis \parencite{tuLogicYeastMetabolic2005}.
As the oxidative phase transitions to the reductive phase, ethanol and acetate concentrations in the medium peak as respiration finishes \citep{tuLogicYeastMetabolic2005}.
`Redox state' metabolites, including NADH, NADPH, glutathione \citep{lloydUltradianMetronomeTimekeeper2005}, and flavins (FMN and FAD)
\parencite{murrayRedoxRegulationRespiring2011} become most oxidised in this phase.
Here, 70\% of metabolite concentrations peak with the combined autofluorescence of NADH and NADPH \citep{murrayRegulationYeastOscillatory2007}.

In the reductive phase, cells consume oxygen at a low rate.
During the reductive-building phase, activities linked to mitochondrial growth occur.
% as shown here.
There is evidence to suggest that activities linked to cell proliferation --- such as initiation of the cell division cycle, DNA replication, and spindle pole activity --- are gated to the reductive-building phase for both the short-period and long-period YMC.
Such evidence includes budding activity and the pattern of the expression of \emph{YOX1}, which encodes a cell division cycle repressor \citep{tuLogicYeastMetabolic2005}.

% There are many strings to pull on this phase.
Finally, during the reductive-charging phase,
non-respiratory metabolism and degradation processes occur to prepare the cell for the oxidative phase.
This non-respiratory metabolism includes glycolysis, ethanol and fatty acid metabolism, and nitrogen metabolism.
With these metabolic modes, under the regulation of the transcription factors Msn2p and Msn4p \citep{kuangMsn2RegulateExpression2017}, acetyl CoA accumulates so ATP can be produced in the oxidative phase \citep{tuLogicYeastMetabolic2005}.
After acetyl CoA levels reach a threshold, it promotes histone acetylation and thus induces the oxidative phase.
These metabolic pathways also optimise production of NADPH --- based on the induction of \emph{GND2} --- to buffer against oxidative stress in the oxidative phase.
Genes associated with protein degradation, ubiquitinylation, peroxisomes, vacuoles, and the proteosome also peak in the reductive-charging phase.

It has been hypothesised that gating activities linked to cell proliferation to the reduce-building phase
creates a temporal separation between oxidative biochemical processes (in OX) and the cell division cycle
This temporal separation may prevent reactive oxygen species generated by oxidative process from damaging DNA.
However, measuring DNA content and oxygen consumption in cells grown at different growth rates \citep{slavovCouplingGrowthRate2011} showed that the S phase of the cell cycle may occur in the oxidative phase if the cells have a slow growth rate.
% Not sure if they changed the growth rate of the cells though -- probably quite difficult
% to do in microfluidics setting.
% Re-phrase it a little to make it more different from Papagiannakis et al.
This may be explained by the YMC gating the early and late cell cycle independently, as evidenced by periodic localisation of the anaphase-promoting complex and mitotic exit activator Cdc14 phosphatase in metaphase-arrested cells \parencite{luPeriodicCyclinCdkActivity2010} and, conversely, the persistence of NAD(P)H cycling upon arresting of the late cell cycle by depletion of Cdc14 \parencite{papagiannakisAutonomousMetabolicOscillations2017}, all observed in single cells.
% What would be the function of such flexible gating?  Seems like community
% has no meaningful consensus yet.  Probably allocation of metabolites.
This evidence indicates that the gating between the YMC and the cell cycle is flexible.
Though there is no meaningful consensus as to the function of such flexible gating,
such gating can be instrumental in the allocation of metabolites to different temporal phases of both cellular oscillators.
% Discuss allocation of metabolites later -- and this can be linked to the modelling chapter of thesis.


\subsection{Yeast metabolic cycles under perturbations}
\label{subsec:intro-ymc-perturbations}
% - Nutrient perturbations
  % - Changing concentration or compositions of carbon sources.
  % - Changing concentration or compositions of nitrogen sources.
  % - Key deletion strains shed light on mechanism

Perturbations in growth conditions can affect the length of the metabolic cycle and its relationship with other cellular events.
The long-phase cycle may vary from 1.4 to 14 hours \citep{caustonMetabolicRhythmsFramework2018}.

\subsubsection{Perturbations in growth conditions}
\label{subsubsec:intro-ymc-perturbations-nutrient}

The main nutrient perturbations people have studied so far are perturbations in carbon sources and in nitrogen sources.

Perturbations in carbon sources are well-documented.
% change glucose concentration
Lower glucose concentrations prolong the metabolic cycle, as evidenced by both chemostat studies that assess the effect of changing the dilution rate \parencite{burnettiCellCycleStart2016, oneillEukaryoticCellBiology2020} and single-cell studies that assess the effect of glucose concentrations in the limiting region \parencite{papagiannakisAutonomousMetabolicOscillations2017}.
% 20 g/L --> 0.5 g/L doesn't seem to significantly prolong it, but maybe
% a greater effect will be seen if we get to the region of glucose limitation
% Though I am aware that this is my experimental observations -- need to see if literature confirms.
For example, decreasing the dilution rate decreases the growth rate, and in turn increases the duration of the oxidative phase relative to the reductive phases \citep{slavovCouplingGrowthRate2011}, thus prolonging the metabolic cycle.
This effect is pronounced in the region of glucose limitation.
Increasing the glucose concentration beyond a certain point does not produce an effect.
Additionally, several studies \citep{slavovCouplingGrowthRate2011,oneillEukaryoticCellBiology2020} show that the duration of the low oxygen consumption phase increases while the duration of the high oxygen consumption phase holds constant if the metabolic cycle duration increases due to these reasons.

Such experimental observations could by explained by models such as \textcite{jonesCyberneticModelGrowth1999}, which suggest that as the dilution rate is decreased, metabolic oscillations acquire greater amplitudes and longer periods, but if it low enough, metabolism becomes entirely oxidative, and metabolic oscillations disappear.
% ferm vs non-ferm
However, non-fermentable carbon sources like pyruvate give long-duration metabolic cycles in single cells, comparable to cells under limiting levels of glucose \parencite{papagiannakisAutonomousMetabolicOscillations2017}.

% bulk addition and depletion
In addition, bulk depletion or addition of a carbon source can reset the phase of the YMC.
Chemostat studies show that an initial starvation phase is needed to generate long-lasting synchronous metabolic cycles \parencite{tuLogicYeastMetabolic2005}. %[OTHER CITATIONS PROBABLY USEFUL].
On the other hand, adding a bulk carbon source such as acetate, ethanol, or acetaldehyde can reset the phase of the YMC \citep{kuangMsn2RegulateExpression2017, krishnaMinimalPushPull2018} or eliminate it \citep{jonesCyberneticModelGrowth1999}.

Perturbations in nitrogen sources are less well-studied.
% Maybe also because long OX stage as it's the stage for biomass building
\textcite{baumgartnerFlavinbasedMetabolicCycles2018}, based on single-cell observations, suggest that decreasing nitrogen source concentration prolongs the YMC, as evidenced by longer flavin oscillations when cells are grown on lower concentrations of yeast nitrogen base (YNB) media or on urea, a non-preferred nitrogen source.

In addition, perturbations outside nutrient sources also affect the YMC.
For example, externally applied hydrogen peroxide, as a source of oxidative stress, shifts the YMC to the oxidative phase \citep{amponsahPeroxiredoxinsCoupleMetabolism2021}.
% [MOVED FROM PREVIOUS SECTION]
Additionally, the oscillation period is insensitive to temperatures from \SI{25}{\celsius} to \SI{35}{\celsius} and media pH values from
2.9 to 6.0 \citep{lloydUltradianMetronomeTimekeeper2005} --- though the period of dissolved-oxygen oscillations decrease as pH decreases to below 2.9 and the oscillations disappear when conditions are too acidic \citep{oneillEukaryoticCellBiology2020}.

Finally, the phase difference between the YMC and cell cycle varies in different conditions % how?
% Confirmation that the YMC responds to nutrient availability?  How does phase
% resetting help the Cell Division Cycle then?
\citep{ewaldYeastCyclinDependentKinase2016}. % ok, but have I seen this phase shift with the different media (i.e. SC vs SM)?  What did Ewald have to say about this phase shift??
% 'bulk carbon source' -- could be adaptation to changing nutrient concentrations in general?
%Furthermore, the YMC can oscillate multiple times per cell cycle, or even disappear in some conditions \citep{baumgartnerFlavinbasedMetabolicCycles2018}. % which conditions?  can these be tested?
% --> also see Causton et al. (2015)
% -----------------------------------------------------------------------------------------------

\subsubsection{Genetic perturbations}
\label{subsubsec:intro-ymc-perturbations-genetic}

Although the molecular basis of the yeast metabolic cycle is not well-characterised, gene deletions shed light on it.
% Mellor (2016) provide an awesome table.
Genes that control the cell division cycle and metabolism have been deleted in studies.

Several deletions have been shown to remove the metabolic oscillations in chemostats: \emph{ZWF1} \citep{tuCyclicChangesMetabolic2007}, \emph{GSY2}, and \emph{GPH1} \parencite{oneillEukaryoticCellBiology2020}.
\emph{ZWF1} codes for glucose-6-phosphate dehydrogenase and is thus responsible for entry into the pentose phosphate pathway and subsequently a major source of NADPH generation, so deleting this gene may impair control of cellular redox.
However, because of its role, this gene deletion impairs adapting to oxidative and pH stress and also causes methionine auxotrophy, so it may be difficult to draw conclusions from this deletion in particular.
Furthermore, other enzyme-catalysed reactions in the cell that generate NADPH exist (Idp2p, Ald6p) and have shown to compensate for the loss of \emph{ZWF1} when cells are grown on lactate plates or on liquid cultures with glucose as the carbon source \parencite{minardSourcesNADPHYeast2005}.
This therefore raises the question of just how important \emph{ZWF1} is to the yeast metabolic cycle, and to what extent is NADPH generation is needed for control of cellular redox.
On the other hand, \emph{GSY2} and \emph{GPH1}, both have roles in glucose/glycogen mobilisation and storage.
The absence of dissolved oxygen cycles in these deletions thus suggest that cycling of carbohydrate stores may be needed for the function of the metabolic cycle.
However, metabolic oscillations have been observed in high-glucose conditions \parencite{papagiannakisAutonomousMetabolicOscillations2017, baumgartnerFlavinbasedMetabolicCycles2018} in which glycogen synthesis is repressed, therefore suggesting that glycogen cycling may play a more minor role in defining the yeast metabolic cycle and another nutrient cycling phenomenon may be more responsible.
In addition, \emph{MSN2} and \emph{MSN4} have been shown to regulate acetyl CoA accumulation in the reductive-charging phase, as evidenced by the lack of YMCs in deletion strains \citep{kuangMsn2RegulateExpression2017}.
This tells us that genes involved in signalling pathways play an important role in the integrity of the metabolic cycle too.

Additionally, other deletions have been shown to change the frequency or shape of dissolved oxygen cycles.
\textcite{caustonMetabolicCyclesYeast2015} provide several examples, of which I discuss \emph{RIM11}, \emph{SWE1}, and \emph{TSA1 TSA2}. % they actually had more, but these are the three that pop in my head right now
Rim11p is the yeast homolog of the GSK3$\beta$ serine/threonine kinase, which regulates metabolism and plays a role in setting the speed of the circadian clock.
The \emph{RIM11} deletion has been shown to result in shortened periods of dissolved-oxygen metabolic cycles in the chemostat, thus pointing towards a common mechanism for both biological oscillators.
Swe1p is a conserved cell division cycle regulator that functions at the G2/M checkpoint and has roles in coupling the cell division cycle with the circadian rhythm.
Deleting \emph{SWE1} also resulted in shortened periods of dissolved-oxygen metabolic cycles but with the same rate of DNA replication, suggesting a dysregulation in the coupling between the yeast metabolic cycle and the cell division cycle.
Tsa1p and Tsa2p are paralogous cytoplasmic thioredoxin peroxidases that cooperate in the peroxiredoxin-thioredoxin system to eliminate reactive oxygen species and have been shown to be a marker for circadian rhythms.
A double deletion of the two associated genes still results in metabolic cycles, but with an additional burst in high oxygen consumption during what would otherwise be the reductive-charging phase, showing that the peroxiredoxin-thioredoxin system is instrumental in the integrity of the yeast metabolic cycle.
In addition, \textcite{amponsahPeroxiredoxinsCoupleMetabolism2021} show the presence of cycling peroxiredoxin oxidation during the YMC using chemostat-based studies, with a corresponding cycling of hydrogen peroxide.
They also confirm that inactivating peroxiredoxins (tsa1$\Delta$ tsa2$\Delta$, additionally with inducible degradation of Ahp1, another cytosolic peroxiredoxin) disrupts the metabolic cycle and decouples it from the cell division cycle.
%Moreover, deletion of \emph{GTS1} shortens the oscillation periods \citep{lloydUltradianMetronomeTimekeeper2005},
Taken together, these deletion studies show that regulators of other biological rhythms and of redox metabolism play a role in the regulation of the YMC.

However, few genetic perturbation studies have been attempted in single-cell studies.
The most significant is in \textcite{baumgartnerFlavinbasedMetabolicCycles2018}, in which by deleting genes (atp5$\Delta$, cyt1$\Delta$) required for respiration, they showed that metabolic cycling does not require respiration.

\subsection{Modelling the yeast metabolic cycle}
\label{subsec:intro-ymc-model}

[THE SUMMARY FIGURES FROM THESE PAPERS MAY BE USEFUL TO HELP ILLUSTRATE THE IDEAS]

Mathematical models have been developed to explain the aspects of the YMC.
An early model is \parencite{jonesCyberneticModelGrowth1999} which simulates dynamic competition between three modes of metabolism --- fermentation, glucose oxidation, and ethanol oxidation --- using differential equations.
This model predicts spontaneous generation of oscillations in dissolved oxygen, cell mass, and storage carbohydrates in continuous cultures.
This prediction is consistent with chemostat-based studies of the yeast metabolic cycle.
Furthermore, the model predicts that, within a window of dilution rate values, if the dilution rate decreases, the dissolved oxygen oscillations increase in amplitude and period.
The increase in period agrees with experimental studies such as \textcite{oneillEukaryoticCellBiology2020}.
However, the model also predicts oscillations in the extracellular concentrations of glucose and ethanol, which conflicts with the steady-state assumption of chemostat studies.

\textcite{krishnaMinimalPushPull2018} uses a frustrated bistability model to produce a relaxation oscillator that explains how a population of yeast cells switches between quiescent and growth states when faced with a limited amount of metabolic resources.
This model assumes that the cells retain hysteresis of their current state and posits that cells of two populations communicate through diffused acetyl-CoA to sustain the population-level oscillatory behaviour.
Read together with the discussion from \textcite{burnettiCellCycleStart2016}, which also propose that yeast cells committed to the metabolic cycle secrete metabolites that induce other cells to enter the metabolic cycle provided that they have enough storage carbohydrates, this model provides an attractive cell-to-cell signalling explanation for the population-level behaviour observed in the chemostat.
However, such an explanation does not explain the presence of metabolic cycling in single-cell conditions in which cells are physically separated and signalling between cells cannot occur -- though it must be noted that autonomous generation of metabolic cycles and synchrony of metabolic cycles in a population can possibly arise from two different mechanisms that are independent of each other.

Based on single-cell experimental observations, \citet{ozsezenInferenceHighLevelInteraction2019} use a deterministic Kuramoto model to explain the interaction between one metabolic oscillator and three cell cycle oscillators at different stages.
This study builds upon use of the Kuramoto model to model collective oscillatory behaviour in other biological systems.
The study uses growth on different carbon source conditions to determine parameters that define the natural frequencies of the cell division cycle oscillators and the strength of the coupling between the four oscillators.
Parameter optimisation predicts that the metabolic cycle most strongly influences the START point of the cell division cycle, and more weakly influences the M and S phases, while the three points of the cell division cycle negligibly influence each other.
Under perturbations, the model system also exhibits stability but also a shift in oscillation frequency, agreeing with experimental observations, and also predicts the effects of Cdc20 and Cdc14 dynamic depletions.
However, a key criticism of this model-based study is that by using the Kuramoto model, it makes simplistic assumptions about the oscillators, which may be unrealistic especially given how little is known about the mechanistic basis of the metabolic oscillator.

% [This is a bit out of place... probably nix it given that I've given up on using this.]
% Additionally, a data-driven stochastic modelling approach was used to describe the relationship between the circadian and cell cycle oscillators without assuming a prior relationship \citep{droin_low-dimensional_2019}.
% This strategy is yet to be applied to the metabolic and cell cycle oscillators.
% However, \citet{ozsezenInferenceHighLevelInteraction2019} assumed the same, specific coupling function for all interactions between the four oscillators.

Taken together, modelling approaches have been able to predict some aspects of the metabolic cycle, but most focus on specific aspects to the detriment of other experimental observations, and none sufficiently reconcile observations from both chemostat-based and single-cell studies.
Constructing more accurate models are complicated by how little of the mechanistic basis of the yeast metabolic cycle has been elucidated thus far.

\subsection{Big picture/Hypothesis: a nutrient sensor than entrains the cell division cycle?}
\label{subsec:intro-ymc-hypothesis}

From existing evidence, we can create a big picture of the yeast metabolic cycle.
The yeast metabolic cycle is a autonomous biological oscillator that operates at a range of frequencies in response to a range of (permissive) growth conditions, as evidenced by how extreme conditions impair the oscillator.
Based on chemostat-based studies, such extreme conditions include poor nutrient quality and media being too acidic.
However, there is reason to believe that the metabolic oscillator can function in some conditions previously deemed to be unfavourable.
For example, single-cell studies show that yeast cell show metabolic oscillations in high-glucose conditions.
Within the permissive growth conditions, different conditions affect the frequency of the metabolic cycle;
for example, a low concentration of glucose or nitrogen source results in longer cycles, and bulk addition of certain compounds can reset the phase of the metabolic cycle.
These observations support the idea that the metabolic oscillator includes the functionality of a nutrient sensor.

The yeast metabolic cycle then creates windows of opportunities for the cell to commit to START if conditions are favourable, for example, good carbohydrate or lipid stores.
Thus, this oscillator acts as a timing mechanism for cellular processes, most importantly the cell division cycle and biosynthetic/redox processes.
Though the relationship between the metabolic cycle and the cell division cycle is governed by the mathematical basis of coupled oscillators.
Most importantly, there is a small window of frequencies in which both oscillators can be phase-locked, and that other, complicated relationships exist: e.g. multiple metabolic cycles per cell division cycle -- in line in the window-of-opportunity idea above.

% **Logical inconsistency**: previously I mentioned that this coupling is flexible &
% this idea is disputed.  This is an error!
% [COMMENTED-OUT MAIN TEXT]
% In turn, if the cell divides, it temporally partitions cell cycle events in relation to YMC events to prevent oxidative damage.

% *** I think chemostat vs single-cell discussions should be introduced WAY earlier, as I can't avoid discussing anything else about the YMC without going into the weeds of this.  But I can then dive into the dispute later.
\subsection{Disputes and unresolved questions with the yeast metabolic cycle}
\label{subsec:intro-ymc-unresolved}
% - Do single-cell flavin-based metabolic cycles from deletion strains recapitulate dissolved oxygen-based metabolic cycles in chemostats?  If not, what would be a likely explanation?

% (State the main unknowns of the yeast metabolic cycle:
% - molecular mechanisms, specifically...? -> deletions to investigate
% - whether things are the same in batch vs bulk)

\subsubsection{Chemostat vs single-cell studies}
\label{subsubsec:intro-ymc-unresolved-chemostat_singlecell}

There is a dispute of whether the same conclusions can be drawn from chemostat-based studies and from single-cell based studies.
Most studies of the YMC arise from chemostat experiments and any conclusion from a single-cell study is subject to the question of whether it recapitulates the YMC in the chemostat.
Reconciling the two types of studies is difficult because the readouts and conditions are different:
chemostat studies produce dissolved oxygen and transcript cycling readings, while single-cell experiments cannot report on dissolved oxygen and chiefly report metabolite cycling.
This leads to differing definitions of the YMC:
some authors \parencite{laxmanBehaviorMetabolicCycling2010, caustonMetabolicRhythmsFramework2018} only use the term metabolic cycle to refer to synchronised cycles of dissolved oxygen concentrations observed in chemostat cultures that must have gone through a starvation phase, while single-cell studies \parencite{baumgartnerFlavinbasedMetabolicCycles2018, zylstraMetabolicDynamicsCell2022} naturally have to expand that definition to include metabolite cycling and sequences of cellular events that are associated with the chemostat metabolic cycle.
For the purposes of my thesis, I adhere to the definition used by single-cell studies.

I argue that there are three caveats to chemostat-based studies:
the experimenter cannot assume that the chemostat is in steady-state,
the chemostat obscures contributions from sub-populations,
and the chemostat imposes glucose starvation.
These caveats affect the interpretation of YMC studies.

There is a wide assumption that the chemostat is in steady-state, but it may not be true.
A mathematical model shows that levels of solutes change over time \citep{jonesCyberneticModelGrowth1999}.
In addition, \citet{oneillEukaryoticCellBiology2020} shows a chemostat setup that promotes evaporation of hydrogen sulfide gas, thus shifting the equilibrium of the reduction of bisulfides.
This may affect redox metabolism in the cell.
In such a case, chemostat observations may not reflect cell-autonomous behaviour.
Instead, the oscillations may reflect individual cells' responses to the initial starvation imposed at the start of chemostat-based studies
The subsequent response to regularly changing media conditions could explain temporal segregation of physiological processes in phases of the YMC.
In other words, the conditions of the chemostat may force the population of cells to behave in a certain way.
However, temporal segregation of physiological process has also been reported in single-cell studies \citep{takhaveevTemporalSegregationBiosynthetic2023}, suggesting that the cycling of solutes in the chemostat could affect some, but not all, temporal aspects of the metabolic cycle.

The chemostat obscures contribution of sub-populations of cells.
\citet{burnettiCellCycleStart2016} suggest that sub-populations within the yeast culture that enter the yeast metabolic cycle in a staggered manner can be responsible for chemostat observations, as evidenced by how the yeast cells spend proportionately more time in the reductive phase at lower dilution rates.
Contributions from sub-populations of cells are further highlighted by \citet{bagameryPutativeBetHedgingStrategy2020}, who used a microfluidic platform to show that there a group of genetically identical yeast cells divide themselves into two populations.
Such a bet-hedging strategy results in some percentage of the population surviving in a glucose-starved or a glucose-rich condition, beneficial for long-term population survival.
Taken together, it is possible that phenotypically different sub-populations in the chemostat culture may partially explain the observations in the chemostat so far.
In addition, there is the question of whether cells individually generate the metabolic cycle or is a diffusible chemical responsible for synchrony, as proposed by \citet{krishnaMinimalPushPull2018}.
Bulk culture set-ups, including chemostats, are not able to address questions about cell sub-populations and autonomy of the metabolic cycle.
However, single-cell set-ups may fill in such a technical gap.

Finally, the chemostat imposes glucose starvation, and single-cell studies with different carbon sources give a different picture in terms of metabolic requirements.
Chemostat studies and related models suggest that glucose starvation and oxidative metabolism are required for oscillations in dissolved oxygen level that define YMCs.
NAD(P)H oscillations have been recorded in non-fermentative conditions, such as pyruvate or low-glucose media \citep{papagiannakisAutonomousMetabolicOscillations2017}.
However, NAD(P)H \citep{papagiannakisAutonomousMetabolicOscillations2017,ozsezenInferenceHighLevelInteraction2019} and flavin \citep{baumgartnerFlavinbasedMetabolicCycles2018} oscillations still occur in constant high-glucose conditions, and only within a window of periods in contrast to the 1.4--14 hour range reported for chemostat-based studies.
Furthermore, \emph{ATP5} and \emph{CYT1} deletions that impair oxidative respiration do not remove single-cell flavin-based metabolic oscillations \citep{baumgartnerFlavinbasedMetabolicCycles2018}, thus giving additional evidence that oxidative metabolism is not required for the YMC.

Single-cell microfluidic studies are well-positioned to address the limitations of the chemostat,
although there have been only few studies.
\citet{laxmanBehaviorMetabolicCycling2010} was an early attempt at using microfluidics to address the bulk vs single-cell issue by culturing strains with fluorescent gene expression reporters for each phase (OX, RB, RC) of the metabolic cycle by transferring cycling cells from a chemostat to a microfluidic device for real-time imaging of the cells.
The study showed that low glucose levels were required for the synchrony of metabolic cycles across cells.
This study also shows the presence of quiescent cells, and then proposed that during OX phase, cells decide whether to commit to cell growth or enter a quiescent state, leading to a model of two subpopulations in the culture.
However, it lacks quantitative time-series analysis, rather, reporting qualitative interpretations of fluorescent images.
The microfluidic device did not truly physically separate each cell individually, thus it is unable to eliminate the possibility of cell-to-cell communication via a diffusible signalling chemical.
% This does seen like a key difference seen in single-cell, but the fact wasn't made clear
% in writing here.
\citet{papagiannakisAutonomousMetabolicOscillations2017} revealed that YMCs are an intrinsic feature of single cells and are autonomous with respect to the cell cycle, based on measurements of the combined level of NADH and NADPH in single cells in microfluidic devices.
Furthermore, by measuring the level of flavins in the cell, \citet{baumgartnerFlavinbasedMetabolicCycles2018} demonstrated that YMCs persist in mutants deficient in oxidative phosphorylation, and that the cell cycle inhibitor rapamycin desynchronises the YMC and the cell cycle.
In sum, these single-cell studies address a small fraction of the knowledge covered by chemostat studies, and further such studies are required.


\subsubsection{Molecular and genetic mechanisms}
\label{subsubsec:intro-ymc-unresolved-molecular}

There are unknowns in the molecular mechanism that drives YMCs.
Genome-wide transcript cycling has two superclusters that correspond to the oxidative and reductive-building phases \citep{machneYinYangYeast2012}.
However, there has been no genome-wide analysis of genes that influence cycling \citep{mellorMolecularBasisMetabolic2016}, though some genes seem to have key roles.
As we lack proteome analysis, it is unclear how protein levels and post-translation modifications are affected.

[FIGURE: MY INVESTIGATION OF CAMPBELL ET AL. 2020 TO SHOW PERIODICITY OF TRANSCRIPTS/PROTEINS/METABOLITES]

Metabolome cycling may play a role in the metabolic cycle and can explain temporal partitioning of biosynthesis, but the evidence so far is indirect as it is based on the cell division cycle.
\textcite{campbellBuildingBlocksAre2020} showed that lipid biosynthesis in budding yeast is periodic with the cell division cycle and peaks during S phase, as evidenced by an increase in the number of metabolites implicated in lipid metabolism in such phases, based on metabolomics analysis of prototrophic cells with synchronised cell division cycles.
Here, precursors are synthesised as they are needed.
\citet{ewaldYeastCyclinDependentKinase2016} also show that the cell division cycle machinery regulate trehalose mobilisation, showing the coupling between carbohydrate store levels and cellular oscillators.
They also showed that lipid metabolism increased during S/G2/M, likely due to the synthesis of new cell membranes during bud growth, as evidenced by pathway enrichment analysis.
Based on the coupling between the yeast metabolic cycle and the cell division cycle, it can be inferred that lipid store cycling and perhaps to a lesser extent carbohydrate store cycling are likely instrumental to the yeast metabolic cycle.
Though, investigation of how an impairment in lipid utilisation affects the yeast metabolic cycle in single-cells is needed to truly prove that such cycles are responsible for the metabolic cycle.
Preferable, I need evidence to support metabolome cycles that also confirms that the YMC is independent of the cell division cycle.

%Additionally, the effects of perturbations to specific metabolic networks in single-cell conditions are yet to be determined [EVIDENCE NEEDED].

\subsection{Implications of the metabolic cycle}
\label{subsec:intro-ymc-implications}

%- campbell et al. 2020: suggests temporal regulation of biosynthesis for cell division cycle -- similar logic to ymc?  can put this in implications of ymc section

Common mechanisms of regulation of the YMC with the cell division cycle and the circadian rhythm
lead to the question of whether the metabolic cycle reflects a fundamental system.

% Describe metabolic cycles in other organisms...
Similar metabolic cycles have been described in other organisms.
\emph{E. coli} shows oscillations in NAD(P)H fluorescence coupled to its cell division cycle, as evidenced by time-lapse microscopy of single cells \citep{zhangDynamicSinglecellNAD2018}.
Addition of glucose or hydrogen peroxide to the medium results in global changes in autofluorescence, reflecting a response to nutrient conditions.
Similar metabolic cycles have been observed in mammalian cells.
For example, \citet{zhuLogicTemporalCompartmentalization2022} describe a 12-hour metabolic cycle in liver cells that includes temporal partitioning of metabolic processes into energy homeostasis, genetic integrity maintenance processes, immune response, and gene expression --- linking the processes to the circadian rhythm and the whole cycle to a more general 12-hour mammalian ultradian clock.
Importantly, this hepatic metabolic cycle operates independently from the spatial organisation of cells in the liver, reminiscent of the autonomy of the yeast metabolic cycle.
In addition, HeLa cells with synchronised cell division have been shown to exhibit both NAD(P)H and ATP oscillations throughout the cell division cycle \citep{ahnTemporalFluxomicsReveals2017}, but the literature is conflicted about the these oscillations' dynamics across different mammalian cell types \citep{zylstraMetabolicDynamicsCell2022}.
There is reason to believe that a wide range of organisms exhibit biochemical phenomena similar to the yeast metabolic cycle, as the aims of controlling cell division to match environmental conditions and temporal coordination of biosynthesis \& cellular redox state with cell division should be fundamental goals that apply to all eukaryotes.

%- Fundamental mechanistic basis for biological oscillators, such as the cell division cycle.  Could change what we know about the cell division cycle.
% (Significance of study)
Metabolic oscillations may be the origins of biological timekeeping mechanisms.
Indeed, \citet{lloydRedoxRhythmicityClocks2007} assert that ultradian oscillations form the basis of longer-period biological oscillators like the circadian rhythm or the cell cycle, based on temperature compensation and sensitivity of the period.
It is logical for a biological oscillation to have temperature compensation because temperature oscillates with a period of a day on most of the planet.
Circadian rhythms can occur in cells of multicellular eukaryotes without transcription \citep{oneillCircadianRhythmsPersist2011}, refuting the idea that gene circuits are responsible for such rhythms.
Additionally, the eukaryotic cell cycle evolved before cyclin-dependent kinases \citep{papagiannakisAutonomousMetabolicOscillations2017}, so metabolic oscillations may have served to regulate the cell cycle before cyclin-dependent kinases evolved.
Furthermore, YMCs share mechanisms with the circadian oscillator \citep{caustonMetabolicCyclesYeast2015,arataQuantitativeStudiesCellDivision2019}, suggesting a common evolutionary origin.
Thus, studying YMCs may shed light on the evolution of biological rhythms.

\section{Flavins and flavoproteins}
\label{sec:intro-flavin}

\subsection{Introduction to cellular autofluorescence}
\label{subsec:intro-flavin-autofluo}
% - (Refer to reviews, e.g. Maslanka et al. 2018 -- is a goldmine)

Cellular autofluorescence is defined as the intrinsic fluorescence of a cell without fluorescent tags.
It is caused by the autofluorescence of compounds that have light emission properties \parencite{maslankaAutofluorescenceYeastSaccharomyces2018}.
Such endogenous fluorophores include coenzymes, vitamins, and amino acids with aromatic chemical groups, flavins being one of them.
However, autofluorescence pose difficulty in cellular microscopy because their wavelengths can overlap with other fluorophores and therefore it is difficult to draw biochemical conclusions from the signal alone.
For example, flavin autofluorescence overlaps with the spectrum of the fluorescent glucose analogue 6-NBDG \parencite{maslankaAutofluorescenceYeastSaccharomyces2018}, and thus interferes with studies that use this analogue to study glucose uptake.
Owing to the identity of the compounds that are responsible for autofluorescence, cellular autofluorescence can indicate of the physiology and metabolism of the cell, and it has been used to study the yeast metabolic cycle.
Autofluorescence thus offers an easy way to monitor cell physiology without engineering genetic constructs.

\subsection{Biochemical basis of flavins and flavoproteins}
\label{subsec:intro-flavin-biochem}

[ADD CHEMICAL DIAGRAM: FLAVIN GROUP, FMN, FAD]

Flavins are a group of organic compounds that share an aromatic moiety that allows redox reactions.
Specifically, the flavin moiety can exist in the oxidised, semiquinone, or reduced states.
Flavins thus function as electron carriers in the cell.
In \emph{Saccharomyces cerevisiae}, flavin is present in FMN and FAD, which function as prosthetic groups in flavin-dependent proteins, or flavoproteins, whose genes account for 1.1\% of the genome \citep{gudipatiFlavoproteomeYeastSaccharomyces2014}.
FMN and FAD can be covalently bound to these proteins or be free.
  % [citation needed -- IIRC they don't exist as the free from for long?  Is the free form transient?].
FAD is a co-enzyme and has major roles in transferring electrons from the TCA cycle to the mitochondrial ETC.

% (Biosynthesis of flavin)
[ADD FIGURE: Schematic of riboflavin uptake and biosynthesis and their relationship with flavin-containing compounds in the yeast cell.  And KEGG diagram.]
% In more detail: https://www.genome.jp/pathway/map00740 (biosynthesis of riboflavin and derivatives, KEGG)

Flavins in \emph{Saccharomyces cerevisiae} are derived from riboflavin.
Riboflavin can be synthesised \emph{de novo} from purine biosynthesis and the oxidative pentose phosphate pathway.
% Is there more to this than the KEGG diagram?
Based on the metabolism of flavins, the cell only synthesises new flavin for synthesis of FMN and FAD.

[ADD FIGURE: SCHEMATIC OF DETECTING FLAVINS VIA MICROSCOPY, THAT I USE IN COUNTLESS POSTERS \& PRESENTATIONS]

[ADD FIGURE(S): FLUORESCENCE EXCITATION \& EMISSION SPECTRUM OF RIBOFLAVIN.  EITHER FROM FPBASE OR ADAPT FROM IN VIVO STUDIES CITED.]

From a technical standpoint, these redox states affect the emission/absorption of electromagnetic radiation of the flavin moiety.

The redox biochemistry of flavins give rise to fluorescence, so monitoring flavin autofluorescence monitors the redox state of the cell.
Flavins, in their oxidised forms (FMN and FAD), have a peak excitation frequency of $\approx$460 nm and a peak emission frequency of $\approx$535 nm \parencite{maslankaAutofluorescenceYeastSaccharomyces2018, wagnieresVivoFluorescenceSpectroscopy1998}.
Comparison of \emph{in vivo} autofluorescence in mammalian cells and the fluorescence spectrum of riboflavin in PBS confirms this fluorescence behaviour \parencite{aubinAutofluorescenceViableCultured1979}.
In contrast, the reduced forms FMNH2 and FADH2 have negligible fluorescence \parencite{mastersConfocalRedoxImaging1994}.
Both chemostat-based \parencite{sasidharanTimeStructureYeastMetabolism2012, murrayRedoxRegulationRespiring2011} and single-cell microfluidic studies \parencite{baumgartnerFlavinbasedMetabolicCycles2018} have monitored flavin autofluorescence to study the YMC.

\subsubsection{Description of key flavoproteins and their roles}
\label{subsubsec:intro-flavin-biochem-descriptions}
% - (Sort by abundance)
% - (On second thought, I'm not sure how useful a shopping list of flavoproteins is.
% probably better to pick a couple that are really key to what i study)

[ADD FIGURE: BAR CHART SHOWING ABUNDANCE OF MOST ABUNDANT FLAVOPROTEINS]

\citet{gudipatiFlavoproteomeYeastSaccharomyces2014} describe 68 genes that code for 47 flavoproteins in budding yeast.
Of these, 35 require FAD, 15 require FMN, and 3 require both.
In budding yeast, most flavins sit in the active site without covalent bonding.
The biochemical and enzymatic properties of many flavoproteins are poorly characterised \citep{kochStructureBiochemicalKinetic2017}.

[ADD TABLE: SHOWING NAMES OF TOP 12 FLAVOPROTEINS, APPROXIMATE ABUNDANCE, THE REACTION THEY CATALYSE, AND THEIR ROLES, AND CITATIONS.  STEAL CONTENT FROM THE FLAVOPROTEINS ORG NOTE]

The most abundant flavoproteins catalyse redox reactions except for Ilv2p.
It has been hypothesised that an ancestral form of Ilv1p catalysed a redox reaction, but the argument is weak because it is inferred from the presence of FAD \citep{pangCrystalStructureYeast2002}.

Specifically, these reactions include reduction of reactive chemical species to respond to oxidative stress --- though not all enzymes involved in the response to reactive chemical species have flavin co-factors.
Additionally, the reactions include biosynthetic reactions.

Many of these reactions require NADPH or NADH to donate electrons, suggesting a link between flavins and NAD(P)H in regulating the cellular redox state.
In particular, Oye2p catalyses the NADPH redox reaction, thus providing a link between flavins and NAD(P)H.
To maintain the cellular redox state, it is thus reasonable to assume that the redox equilibrium of all flavoprotein-catalysed reactions are in the same direction at any point of the YMC.
Supporting this, \citet{sianoNADHFlavinFluorescence1989} show that NAD(P)H fluorescence and the fluorescence of lipoamide dehydrogenase, a flavoprotein, indicate simultaneous reduction in response to lowered dissolved oxygen.
They further show redox equilibrium in both fluorophores in response to glucose addition.

It is important to rule out the possibility that flavin cycling is merely a function of the cell division cycle to make sure that flavin monitoring monitors the YMC.
None of these flavoproteins are strictly cell division cycle proteins, but this does not preclude cell division cycle-linked cycling of flavin autofluorescence.
For example, fatty acid synthesis proteins should cycle along with the cell division cycle as cell synthesises more plasma membrane.

\subsection{Flavins and flavoproteins in the yeast metabolic cycle}
\label{subsec:intro-flavin-ymc}

Flavin fluorescence can be used to monitor the metabolic cycle.
The biological basis of flavins justifies this use.
Flavins are linked to NAD(P)H via nitric oxide oxidoreductase (Yhb1), as discussed in section \ref{subsubsec:intro-flavin-biochem-descriptions}, and NAD(P)H cycles have been implicated in bulk-culture \citep{tuLogicYeastMetabolic2005} % and more
and single-cell \citep{papagiannakisAutonomousMetabolicOscillations2017} studies of the YMC, as discussed in section \ref{subsubsec:intro-ymc-definition-phases}.
The timing of flavin cycle peaks -- indicating oxidation -- coincides with the oxidative state of the YMC, as evidence by how flavin fluorescence peaks in-phase with oxygen uptake rates in the chemostat \citep{murrayRedoxRegulationRespiring2011,sasidharanTimeStructureYeastMetabolism2012}.
Riboflavin has been shown to oscillate and peak in the oxidative state of the YMC, while FAD peaks in the reductive-building phase, as evidenced by metabolic profiling of extracts from chemostat cultures taken at evenly-spaced intervals \parencite{tuCyclicChangesMetabolic2007}.
Most abundant flavoproteins may have roles linked with the YMC.
The most abundant is Fas1 (fatty acid synthetase); because there is evidence that cycles of fatty acid stores are implicated in metabolic cycling in yeast \citep{campbellBuildingBlocksAre2020}, it is likely that fatty acid synthetase is heavily implicated.
Following this, the second most abundant is Yhb1, which may play a major role as discussed earlier.

So, for these reasons, I expect flavin autofluorescence to be oscillatory and be a useful read-out of the yeast metabolic cycle.
Few studies have characterised how such oscillations respond to changing nutrient conditions or to gene deletions.
Thus, filling in this knowledge gap is an avenue for further research.

% [MARKED FOR REORGANISATION]
Nevertheless, there are caveats of using flavin.
There could be other things fluorescing in that wavelength region, although the benefits of using a non-invasive method to monitor the redox metabolism of the cell that doesn't require additional genetic engineering outweighs the caveats.
And, this applies to other sources of autofluorescence too, including NAD(P)H.
Additionally, judging by flavin measurements alone, we cannot be sure if changes in the reading are due to changes in the `flavin pool' (the amount of flavin-derived moieties in the cells across all their redox states) or due to global changes in intracellular flavin redox state, as a function of intracellular redox state.
KEGG pathway (see section ...) suggests that there is a mechanism for the cell to increase its flavin pool by uptake of riboflavin from the media or from biosynthesis; certainly, the uptake route via Mch5p is confirmed by \textcite{gudipatiFlavoproteomeYeastSaccharomyces2014}.
However, the experimenter may reduce the chance of the former possibility by using minimal media that does not contain riboflavin, which is a common component in yeast media.

% This paragraph is asking to be deleted because it repeats stuff from earlier.
Most studies [CITATIONS NEEDED] assume a constant flavin pool, based on [DISCUSS THEIR EVIDENCE HERE].
And therefore the oscillations can be seen as periodic shifts in redox equilibrium.
Furthermore, measuring flavin fluorescence also incorporates fluorescence from riboflavin, in addition to flavoproteins.
Either way, flavin fluorescence is the sum of the activity of many components, and this leads to another caveat: you cannot conclude that flavin fluorescence is mostly Fas1 or any other protein, and the only biochemical conclusion you can make is conclusions about overall cellular redox state.
To close, I have to note that some of these caveats are not unique to flavin fluorescence, but a shared limitation of using fluorescence.


MASLANKA
Flavin fluorescence is less sensitive to environmental conditions than NAD(P)H fluorescence, which requires strictly controlled conditions for measurement.
Different concentrations of riboflavin influence the autofluorescence signal and also the physiological status of cells, as an antioxidant, and as a precursor to FAD which plays a role in maintaining the levels of GSH.
% [END MARK]

%%% Local Variables:
%%% mode: latex
%%% TeX-master: "../thesis.tex"
%%% End:
