\chapter{Modelling yeast biosynthesis strategies under constraints}
\label{ch:model}

To better understand the mechanistic basis of the YMC, I sought to build a model of how the cyclic sequence of cellular events respond to extracellular nutrient conditions.
However, it is challenging to develop a fine-grained model for the aspects of the YMC,
especially if the detailed molecular mechanisms are unclear.
Further complicating the development of a fine-grained model is the fact that the main read-outs from single-cell studies, NAD(P)H and flavin autofluorescence, are aggregate signals from several biochemical phenomena.
Thus, it is more feasible to construct coarse-grained models to answer biological questions about the YMC.

Here, I use a genome-scale metabolic model and flux balance analysis (FBA) to address whether cellular events in the metabolic cycle reflect a strategy in resource allocation that optimises growth.

Specifically, I aim to evaluate these hypotheses:
\begin{enumerate}
  \item A finite proteome pool gives rise to sequential scheduling of the synthesis of biomass components during growth: lipid, carbohydrate, amino acids, and nucleic acids.
        In other words, as an adaption, the cell synthesises biomass components in sequence rather than in parallel.
        This would explain the timing of biosynthetic events in the phases of the yeast metabolic cycle.
  \item This resource allocation strategy remains advantageous in many nutrient conditions and deletion strains,
        which would explain the robustness of the yeast metabolic cycle.
  \item Synthesis of biomass components in parallel, as opposed to in sequence, may be advantageous if the processes of synthesising each biomass component share similar enzyme levels across biomass components.
\end{enumerate}

\section{Introduction to flux balance analysis}
\label{sec:model-fba}

Metabolic network reconstructions are mathematical representations of a set of metabolic pathways in a living organism.
Usually, each metabolite is represented as a node, and metabolites are connected to each other through reactions that are represented as links \parencite{palssonSystemsBiologyConstraintbased2015}.
This information can be represented in a two-dimensional stoichiometric matrix, in which the rows of the matrix represent the metabolites, the columns represent the reactions, and the values of each element in the matrix show the stoichiometry of the reactions in the system.

For example, if reaction $R_{1}$ is defined by:

\begin{equation}
  \ce{1 M1 + 2 M2 -> 3 M3}
  \label{eq:model-example-chemical-reaction}
\end{equation}

the elements in the stoichiometric matrix that correspond to the metabolite-reaction combinations $(M_{1}, R_{1})$, $(M_{2}, R_{1})$, and $(M_{3}, R_{1})$ are -1, -2, and 3, respectively.

A genome-scale metabolic model is, in simple terms, a metabolic network reconstruction that aims to cover every biochemical reaction in a living system that is catalysed by a gene-encoded enzyme.
In most of the reactions represented in a metabolic network reconstruction, one or more chemical species react to create a different set of chemical species as products.
However, reactions in a metabolic network reconstruction may include processes that are not chemical reactions.
Such processes may include exchange for nutrients, as well as a reaction that models biomass formation.

Flux balance analysis (FBA) is a mathematical method that finds the steady-state flux of reactions through a metabolic network that is best for a given condition \parencite{orthWhatFluxBalance2010}.
These metabolic fluxes represent rates of chemical reactions.
At its core, FBA is a method of solving a linear programming problem of finding the flux values that optimise the output value of an objective function, subject to biological constraints.

Mathematically, the linear programming problem of FBA can be expressed as:

\begin{equation}
  \max \mathbf{c}^{\intercal} \mathbf{v}
  \label{eq:model-fba-objective}
\end{equation}

subject to

\begin{equation}
  \begin{gathered}
    \mathbf{S} \mathbf{v} = \mathbf{0}\\
    v_{i,\mathrm{min}} \leq v_{i} \leq v_{i,\mathrm{max}}
  \end{gathered}
  \label{eq:model-fba-constraints}
\end{equation}

where $\mathbf{c}$ is a vector of weights such that $\mathbf{c}^{\intercal} \mathbf{v}$ defines the objective function, $\mathbf{S}$ is the stoichiometric matrix, and $\mathbf{v}$ is the vector of fluxes. The expression $v_{i,\mathrm{min}}$ the lower bound and $v_{i,\mathrm{max}}$ represents the upper bound for each flux $v_{i}$ in $\mathbf{v}$.

The objective function is given as the task of maximising a mathematical expression that is based on a subset of fluxes (equation~\ref{eq:model-fba-objective}).
Most commonly, the objective function for an FBA problem is maximising the flux of the biomass reaction, thus optimising the growth rate of the cell.
The constraints for FBA are, in the most basic case, imposed by two factors:
the stoichiometric matrix and reaction flux bounds (equation~\ref{eq:model-fba-constraints}).
The stoichiometric matrix balances reaction inputs and outputs, while flux bounds impose upper and lower limits on the fluxes of each reaction.
These constraints restrict the solution space for the FBA problem.

FBA thus offers a computationally inexpensive way to simulate metabolism in a living system, as opposed to solving a large set of differential equations that describe the kinetics of biochemical reactions that are difficult to construct and parametrise.

\section{Modelling temporal scheduling of biosynthesis}
\label{sec:model-temporal}

Previous studies have attempted to use FBA to model how each phase of the YMC has different metabolic requirements.
\textcite{takhaveevTemporalSegregationBiosynthetic2023} showed that in different stages of the cell division cycle, the cell synthesises different components of its biomass at different levels.
In the study, they blocked synthesis of each class of macromolecule and recorded the changes in single-cell NAD(P)H cycles, representing the YMC, to quantify the level of each class of macromolecule that the cell synthesises at each time point within a cell division cycle.
Then, they used these activities as coefficients for a modified thermodynamic-stoichiometric metabolic model at each time point and used FBA to deduce biomass production rates.
% Does not suggest that synthesis of each class of macromolecule excludes all others (which makes sense).
% Rather, this study confirms that timescale of observed synthesis events matches simulations.
% Do I even cite this?  It's so bad.
Additionally, \textcite{cesurGenomeWideAnalysisYeast} constructed a different FBA model for each YMC phase based on transcriptomic and epigenetic data.
They did so using the GIMME algorithm \parencite{beckerContextSpecificMetabolicNetworks2008}, which excludes reactions that correspond to genes that are not expressed at certain time points.
Both studies model the metabolic state of the cell at each phase of the YMC, rather than predicting the time the cell takes to replicate or to synthesise biomass components.

An attempt in extending FBA to solve a time-dependent resource allocation problem was \textcite{reimersCellularTradeoffsOptimal2017}.
This study extended a genome-scale model of the cyanobacterium \textit{Synechococcus elongatus} PCC7942 to find the temporal order of intracellular synthesis reactions which optimises the growth rate of the cell, under resource constraints.
However, the model relies on an external oscillator --- namely, the light-dark cycle.
In contrast, the yeast metabolic cycle is autonomously generated.
Therefore, it is difficult to extend the ideas in \textcite{reimersCellularTradeoffsOptimal2017} to the study of the YMC.

Traditional genome-scale models assume that the uptake rate of carbon source limits production.
However, levels of each enzyme also restrict reaction fluxes, leading to the development of enzyme-constrained models.
An enzyme-constrained model fits the assumption that there is a fixed number of amino acids the cell has to distribute \parencite{weisseMechanisticLinksCellular2015}.
Models like \textcite{sanchezImprovingPhenotypePredictions2017} and \textcite{elsemmanWholecellModelingYeast2022} constrain the total sum of fluxes based on a defined total amount of enzyme.
\textcite{elsemmanWholecellModelingYeast2022} additionally impose a ribosome capacity constraint and additionally imposes compartment constraints.

In this chapter, I used an enzyme-constrained genome-scale model of \textit{Saccharomyces cerevisiae} and performed FBA to simulate temporal scheduling of biomass components.
To simulate the cell prioritising the synthesis of each biomass component, I ablated metabolites from the biomass reaction so that one biomass component remained, then optimised the model.
To assess the advantage of sequential synthesis of biomass components over parallel synthesis of biomass components, I estimated the timescales for the synthesis of each biomass component based on the ablations, and compared them to timescales estimated based on the unmodified biomass reaction.
To evaluate the hypothesis that a restricted proteome pool gives rise to sequential synthesis of biomass components, I varied the size of the proteome pool to observe its effects on the cell's preferred resource allocation strategy.
Finally, to show how nutrient conditions affect the cell's preferred resource allocation strategy, I modelled changes in carbon and nitrogen source concentrations and investigated their effects on the quantities that explain the strategies.

\section{The Yeast8 and ecYeast8 models and their formalisms}
\label{sec:model-yeast8}

To impose proteome constraints, I use the enzyme-constrained Yeast8 (ecYeast8) model \parencite{luConsensusCerevisiaeMetabolic2019} and performed FBA.
I use this model because it is recent, offering a good coverage of reactions, and is continuously updated in a well-characterised and well-documented software repository.
Here, I use the model ec\-Yeast\-8.6.0, the latest version for which both original and enzyme-constrained variants are available.
ecYeast8 uses the GECKO formalism \parencite{sanchezImprovingPhenotypePredictions2017} --- specifically, GECKO 2 \parencite{domenzainReconstructionCatalogueGenomescale2022}, the latest published version of GECKO.
GECKO applies an enzyme constraint by modifying the stoichiometric matrix of a genome-scale metabolic model.
In addition, the model has `pseudometabolites' defined by reactions that group specific chemical species in general classes.
These formalisms allow studying each class of biomass component (e.g. lipid, protein, carbohydrate) individually.

In this section, I discuss the formalisms used in ecYeast8 because they differ from the usual formalisms used in genome-scale metabolic models.

ecYeast8 contains four formalisms relevant to this chapter:
\begin{enumerate}
  \item
        The biomass reaction is defined by having `pseudometabolites' as reactants and a biomass species as a product.
        These pseudometabolites include lipids, proteins, carbohydrates, RNA, DNA, ions, and cofactors.
        This is in contrast to BiGG genome-scale models \parencite{norsigianBiGGModels20202020}.
        In these models, biomass reactions have chemical species as reactants, and each has a stoichiometric coefficient that is equal to the species' abundance in units of \SI{}{\mmolgdw}.
  \item
        The pseudometabolites are defined by `\textit{isa}' reactions, which group specific chemical species into classes of metabolites \parencite{heavnerYeastExpandedReconstruction2012}.
        These reactions account for how some KEGG definitions of reactions require generic compounds and these reactions also allows flexibility of biomass definition in different growth conditions.
        The \textit{isa} reactions have chemical species as reactants, each with a stoichiometric coefficient representing the species' abundance in units of \SI{}{\mmolgdw}, and a pseudometabolite as a product.
        In effect, the species abundance information is shifted one reaction away from the biomass reaction.
  \item
        The models implements SLIMEr \parencite{sanchezSLIMErProbingFlexibility2019}, which Splits Lipids Into Measurable Entities, and adds constraints on lipid classes and acyl chain distribution.
        This formalism is needed because species of lipid backbones and acyl chain can combine to form lipids in more than a thousand ways, and the resulting lipid species are difficult to all be represented in a genome-scale metabolic model.
        SLIMEr thus introduces reactions that split lipids into their basic components and lipid pseudoreactions to preserve the distribution of acyl chains.
        As a result, the definitions of lipids are flexible.
  \item
        GECKO was applied to Yeast8 to produce ecYeast8.  GECKO modifies the stoichiometric matrix of an input genome-scale metabolic model to account for enzyme abundances and kinetics.
        Specifically, it adds to enzyme-catalysed reactions enzyme species with a stoichiometric coefficient derived from its $k_{cat}$ value.
        The formalism also adds reactions to model drawing enzymes from a pool.
        GECKO simulates an upper limit of amino acids available for enzyme production.
\end{enumerate}

Here, I detail how I apply these formalisms for this chapter.
% TODO: Use proper refs rather than hard-coding numbers.
In particular, I detail GECKO (item 4) and computing molecular weights of pseudometabolites (based on items 1 and 2).


\subsection{GECKO}
\label{subsec:model-yeast8-gecko}

\begin{figure}
  \centering
  \includegraphics[width=0.9\linewidth]{sanchezImprovingPhenotypePredictions2017_1b_adapted}
  \caption{
    Modifications to the stoichiometric matrix of a genome-scale model that GECKO imposes.
    The original stoichiometric matrix $\mathbf{S}$ includes metabolites $M_{1} \ldots M_{m}$ and reactions $v_{1} \ldots v_{n}$.
    GECKO extends this matrix to include enzymes $E_{1} \ldots E_{p}$ and enzyme usage reactions $e_{1} \ldots e_{p}$.
    The new stoichiometric matrix can be seen as four submatrices concatenated together: the upper right submatrix is $\mathbf{0}$, the lower left submatrix encodes kinetic information, and the lower right submatrix is $\mathbf{I}$.
    Figure adapted from~\textcite{sanchezImprovingPhenotypePredictions2017}.
  }
  \label{fig:model-gecko}
\end{figure}

In a conventional genome-scaled model, metabolic fluxes through reactions are constrained by lower and upper bounds.
This constraint narrows down the solution space when the objective function is optimised.
GECKO imposes an additional constraint on the metabolic fluxes based on the concentration of the enzyme that catalyses the reaction (figure~\ref{fig:model-gecko}).

GECKO modifies the linear programming of FBA as defined by equations~\ref{eq:model-fba-objective} and~\ref{eq:model-fba-constraints} so that enzymes are expressed as metabolites that take part in reactions.
In mathematical terms, this can be expressed as:

\begin{equation}
  \max \mathbf{c}^{\intercal} \mathbf{v}
  \label{eq:model-gecko-fba-objective}
\end{equation}

subject to

\begin{equation}
  \begin{gathered}
    \mathbf{S} \mathbf{v} = \mathbf{0}\\
    v_{j,\mathrm{min}} \leq v_{j} \leq v_{j,\mathrm{max}}\\
    v_{j} \leq k_{\mathrm{cat}}^{ij} \cdot [E_{i}]
  \end{gathered}
  \label{eq:model-gecko-fba-constraints}
\end{equation}

where $v_{j}$ is the flux of each reaction $R_{j}$ catalysed by enzyme $E_{i}$, $k_{\mathrm{cat}}^{ij}$ is the catalytic constant of the enzyme $E_{i}$ for reaction $R_{j}$, and $[E_{i}]$ represents the concentration of the enzyme $E_{i}$.
Defining constraints in this way means that each $v_{j}$ must not exceed $v_{\mathrm{max}}$ for each enzyme $E_{i}$.
Each enzyme $E_{i}$ may catalyse one or more reactions $R_{j}$.

% The constraints in Eq.\ \ref{eq:model-gecko-fba-constraints} is imposed by extending the stoichiometric matrix $\mathbf{S}$ and flux vector $\mathbf{v}$.
% Specifically, $\mathbf{S}$ has additional rows to represent new metabolites that represent each enzyme $E_{i}$.
% The matrix $\mathbf{S}$ also has additional columns for new reactions $ER_{i}$ that model enzyme usage and for a new reaction $ER_{\mathrm{pool}}$ that models a limited proteome pool.
% Kinetic information in the form of $k_{\mathrm{cat}}^{ij}$ values are included as elements in the extended stoichiometric matrix.
% Finally, $\mathbf{v}$ has additional elements that correspond to the additional columns of $\mathbf{S}$.

% The constraint $v_{j} \leq k_{\mathrm{cat}}^{ij} \cdot [E_{i}]$ from Eq.\ \ref{eq:model-gecko-fba-constraints} can be expressed as:

% \begin{equation}
%   \begin{gathered}
%     -\frac{1}{k_{\mathrm{cat}}^{ij}}v_{j} + e_{i} = 0\\
%     0 \leq e_{i} \leq [E_{i}]
%   \end{gathered}
%   \label{eq:model-gecko-fba-constraints-extended}
% \end{equation}

% where $e_{i}$ represents the flux of an enzyme usage reaction $ER_{i}$ associated with $E_{i}$.

To illustrate what the modified stoichiometric matrix means for each reaction, consider a simple example, a reaction $R_{j}$ catalysed by enzyme $E_{i}$:

\begin{equation}
  \ce{ A + B ->[E_{i}] C + D }
  \label{eq:model-gecko-catalysed}
\end{equation}

To apply the constraints described by Eq.\ \ref{eq:model-gecko-fba-constraints}, the chemical reaction in Eq.\ \ref{eq:model-gecko-catalysed} is modified by adding a term:

\begin{equation}
  \ce{ n_{ij}E_{i} + A + B -> C + D }
  \label{eq:model-gecko-catalysed-formalism}
\end{equation}
with the stoichiometric coefficient $n_{ij} = 1/k_{\mathrm{cat}}^{ij}$.
GECKO takes $k_{cat}$ values from BRENDA \parencite{changBRENDAELIXIRCore2021}.

This transformation, adding the enzyme as a pseudometabolite, is based on the intuition that the system uses some amount of enzyme at a specific time to catalyse the flux going through the reaction.
%
Slightly different formalisms are applied to reversible reactions, isozymes, promiscuous enzymes, and enzyme complexes, but these are beyond the scope of this chapter.
% Namely:
% \begin{itemize}
%   \item Reversible reactions are modelled as the forward and reverse reactions separately.
%   \item For isozymes, the original reaction is copied multiple times corresponding to the number of reactions that the isozyme catalyses.
%         Each has an isozyme catalysing the reaction.
%         In addition, there is an `arm' reaction to act as an intermediate between the substrate and the products.
%   \item No actions are needed for promiscuous enzymes.
%   \item Enzyme complexes are modelled as one reaction that uses all subunit proteins that all share the same $k_{cat}$ value.
% \end{itemize}

To constrain overall enzyme levels in the model, GECKO defines a pseudoreaction

\begin{equation}
  ER_{\mathrm{pool}}: \varnothing \ce{ -> E_{pool} }
  \label{eq:model-gecko-enzyme-pool}
\end{equation}

where $E_{\mathrm{pool}}$ is a pseudometabolite that represents the proteome pool available for enzymes.
This pseudoreaction has a flux

\begin{equation}
  \epool \leq (P_{\mathrm{total}} - P_{\mathrm{measured}}) \cdot f \cdot \sigma
  \label{eq:model-gecko-enzyme-pool-flux}
\end{equation}

in units of \SI{}{\gram~\gram_{DW}^{-1}}, where
$P_{\mathrm{total}}$ is the total protein fraction with respect to the dry weight of the cell,
$P_{\mathrm{measured}}$ is the protein fraction of proteins whose weight are accounted for in the model, based on proteomic data,
$f$ represents the fraction of proteins that are enzymes, and
$\sigma$ is a parameter that represents the average saturation of enzymes.
ecYeast8.6.0 assumes parameter values of: $f = 0.5$, $P_{\mathrm{total}} = 0.5$, and $\sigma = 0.5$.
If no proteomic data is used, as is the case in this chapter, $P_{\mathrm{measured}} = 0$.

Defining such parameters is a judgement call, especially when the protein fraction varies across growth rates~\parencite{elsemmanWholecellModelingYeast2022}, but $f = 0.5$ is close to the protein mass fraction of ecYeast8.6.0.
Subsequently, GECKO changes the carbohydrate composition based on the assumption that a change in the amino acid composition is offset by the reverse change in the carbohydrate composition;
experimental data justifies this assumption \parencite{nissenFluxDistributionsAnaerobic1997}.

Then, for each enzyme $\ce{E_{i}}$, GECKO defines enzyme usage pseudoreactions of the form

\begin{equation}
  ER_{i}: \ce{ MW_{i} E_{\mathrm{pool}} -> E_{i} }
  \label{eq:model-gecko-enzyme-usage}
\end{equation}

where $\mathrm{MW}_{i}$ represents the molecular weight of the enzyme in units of \SI{}{\gram~\milli\mole^{-1}}.
The flux of enzyme usage pseudoreactions are defined in units of \SI{}{\mmolgdw}.
GECKO takes enzyme data from SWISSPROT \parencite{theuniprotconsortiumUniProtUniversalProtein2023} and KEGG \parencite{kanehisaKEGGTaxonomybasedAnalysis2023}, including molecular weight of proteins and associated pathways.


Taken together, the modelled cell thus has an enzyme pool in terms of a mass fraction of the cell's dry weight, and the modelled cell allocates certain fractions of this mass to the synthesis of each enzyme at steady-state.
The mass of each enzyme in the cell determines the amount (in moles) of each enzyme and therefore its catalytic activity.


\subsection{Computing molecular weights of pseudometabolites}
\label{subsec:model-yeast8-molweights}

To compute the time it takes for the yeast cell to synthesise each biomass component, the mass fractions of each biomass component must be known.
However, the mass fraction of each biomass component varies according to strain and growth rate \parencite{nilssonMetabolicTradeoffsYeast2016, elsemmanWholecellModelingYeast2022}.
Therefore, I computed the mass fractions of each biomass component based on their stoichiometric coefficients in the ecYeast8 model (appendix~\ref{append:model-molweights}).

\begin{table}[ht]
  \centering
  \begin{tabular}{lSS}
    Metabolite & {\makecell{Computed molecular weight\\ (\SI{}{\gram~\mol^{-1}})}} & {\makecell{Biomass composition\\ at growth rate \SI{0.375}{\hour^{-1}}\\ (\SI{}{\gram~\kilo\gram_{DW}^{-1}})}} \\
    \hline
    Protein & 504.37 & 505.\\
    Carbohydrate & 350.37 & 237.\\
    RNA & 64.04 & 105.\\
    Lipid & 31.57 & 57.\\
    Cofactors & 4.83 & \\
    DNA & 3.90 & 5. \\
    Ions & 2.48 & \\
    \hline
    Total & 961.57 & \\
  \end{tabular}
  \caption{
    Computed molecular weights of bulk metabolites in ecYeast8, compared to experimentally recorded biomass composition by \textcite{canelasVivoDatadrivenFramework2011}.
  }
  \label{tab:ecyeast8-mol-weights}
\end{table}

Table~\ref{tab:ecyeast8-mol-weights} summarises the molecular weight of the pseudometabolites.
As validation of the calculations of molecular weights, the ratio between the molecular weights are similar to the ratio between the mass of each class of macromolecule in the yeast cell dry weight shown by \textcite{canelasVivoDatadrivenFramework2011}.

Adding together values in table~\ref{tab:ecyeast8-fatty-acid-backbone-rxns} gives a molecular weight of the biomass pseudometabolite of \SI{961.57}{\gram~\mol^{-1}}, close to the \SI{966}{\gram~\mol^{-1}} computed by \textcite{takhaveevTemporalSegregationBiosynthetic2023} from a different genome-scale model.
In theory, this number should be \SI{1000}{\gram~\mol^{-1}} because the stoichiometric coefficients of the species that form biomass components are expressed in terms of \SI{}{\mmolgdw} \parencite{thieleProtocolGeneratingHighquality2010, palssonSystemsBiologyConstraintbased2015}, but the deviation from 1000 could be explained by the SLIMEr formalism.
Plus, the sum of stoichiometric coefficients are not always verified in genome-scale models \parencite{chanStandardizingBiomassReactions2017}.


\section{Ablating pseudometabolites from the biomass reaction}
\label{sec:model-yeast8-pseudometabolites}

\subsection{Definition of ablation}
\label{sec:model-yeast8-pseudometabolites-def}

To simulate producing each class of biomass component in turn,
I take advantage of pseudometabolites in ecYeast8 to remove each of them in turn from the biomass reaction ---
in other words, ablating pseudometabolites from the biomass reaction.
Specifically, I exclude each pseudometabolite from the biomass reaction, in turn, and optimise the model.

Consider the objective function, the biomass reaction:

\texttt{
  47.5883 atp\_c + 47.5883 h2o\_c + lipid\_c + protein\_c + carbohydrate\_c\\
  + dna\_c + rna\_c + cofactor + ion \\
  --> 47.5883 adp\_c + biomass\_c + 47.5883 h\_c + 47.5883 pi\_c
}

There are seven pseudometabolites: lipid, protein, carbohydrate, DNA, RNA, cofactor, and ion.

To have the cell prioritise biosynthesis of lipids, I set the stoichiometric coefficients of all pseudometabolites except for lipids to zero in the above equation, giving:

\texttt{
  47.5883 atp\_c + 47.5883 h2o\_c + lipid\_c \\
  --> 47.5883 adp\_c + biomass\_c + 47.5883 h\_c + 47.5883 pi\_c
}

The model was optimised using FBA, with the modified reaction as the objective function.
This process was repeated for the other pseudometabolites.

\begin{figure}
  \centering
  \includegraphics[width=.6\linewidth]{ablation_example_fluxes.png}
  \caption{
    %Growth rates from the original model ($\gro$; leftmost bar) and from the ablated versions of the model ($\griabl$, where $i$ represents each biomass components among lipids, proteins, carbohydrates, DNA, RNA, cofactors, and ions; other bars)
    Growth rates from the original model (leftmost bar) and from the ablated versions of the model (other bars)
  }
  \label{fig:model-ablate-fluxes}
\end{figure}


\subsection{Effect of ablation on allocation of proteome to enzymes}
\label{sec:model-yeast8-pseudometabolites-allocation}

To evaluate whether ablation leads to changes of model behaviour that reflect cellular biochemistry, I quantified how proteome allocation to enzymes changes across rounds of ablation.
When the model prioritises a different biomass component in each round of ablation, the model partitions the limited proteome available for enzyme production differently.
In other words, in each round of ablation, the vector of fluxes carried by each enzyme usage pseudoreaction (defined in Eq.\ \ref{eq:model-gecko-enzyme-usage}) is different from each other and from the non-ablated case, in which all biomass components are synthesised in parallel.
Optimising the objective function as each pseudometabolite is prioritised affects the predicted growth rates (Fig.\ \ref{fig:model-ablate-fluxes}).

Subsystem information aids interpretation of these changes.
Here, I match the flux carried by each enzyme usage pseudoreaction to the enzyme-catalysed reaction that the enzyme is associated with.
If an enzyme usage pseudoreaction is associated with multiple enzyme-catalysed reactions, data entries were duplicated accordingly.
Then, the subsystem associated with each enzyme-catalysed reaction is taken from the gene-protein map, and noted.

To quantify changes in proteome allocation across rounds of ablation, I compute the $\log_{2}(\mathrm{FC})$ of fluxes relative to the non-ablated, parallel case.
This is defined as:

\begin{equation}
  \log_{2}(\mathrm{FC}_{i,j}) = \log_{2}\left( \frac{e_{i,j}^{\prime}}{e_{i, \mathrm{par}}^{\prime}} \right)
  \label{eq:model-foldchange}
\end{equation}

where, to ensure that $\log_{2}(\mathrm{FC}_{i,j})$ can be defined for all $i$ and $j$,

\begin{equation}
  e^{\prime} =
  \begin{cases}
    \epsilon, & \text{if}\ |e|<\epsilon \\
    e, & \text{otherwise}
  \end{cases}
  \label{eq:model-epsilon-round}
\end{equation}

where $e$ is either $e_{i,j}$ or $e_{i, \mathrm{par}}$.
Here, $e_{i, \mathrm{par}}$ represents the flux of an enzyme usage reaction associated with each enzyme $E_{i}$ in the model (defined in Eq.\ \ref{eq:model-gecko-enzyme-pool}) in the non-ablated, parallel case, and $e_{i,j}$ represents the flux of the enzyme usage reaction when biomass component $j$ is prioritised in a round of ablation.

The value $\epsilon$ is defined as:

\begin{equation}
  \begin{aligned}
    \epsilon &= \SI{1}{\molecule~\cell^{-1}~\hour^{-1}}\\
             &= \frac{(\SI{1}{\molecule})}{(\SI{1}{\cell})(\SI{1}{\hour})} \cdot \frac{(\SI{1}{\cell})}{(\SI{15}{\pico\gram_{DW}})} \cdot \frac{(\SI{1}{\mol})}{(\SI{6.02214d23}{\molecule)}}\\
             &= \SI{1.11d-13}{\mol~\gram_{DW}^{-1}~\hour^{-1}}\\
             &= \SI{1.11d-10}{\mmolgdwh}\\
  \end{aligned}
  \label{eq:model-epsilon}
\end{equation}

assuming a reasonable minimum reaction flux of \SI{1}{\molecule~\cell^{-1}~\hour^{-1}} and a cell dry weight of \SI{15}{\pico\gram} dry weight per cell \parencite{shermanGettingStartedYeast2002}.

If $\log_{2}(\mathrm{FC_{i,j}}) > 0$, this indicates that the cell allocates more of its proteome to produce enzyme $E_{i}$ when biomass component $j$ is prioritised.
Conversely, if $\log_{2}(\mathrm{FC_{i,j}}) < 0$, the cell allocates less of its proteome to produce enzyme $E_{i}$ when biomass component $j$ is prioritised.
In addition, if there is a case of enzyme expression switching on, evidenced by $e_{i, \mathrm{par}} = 0$, $\log_{2}(\mathrm{FC}_{i,j}) \ll 0$.
Conversely, if there is a case of an enzyme expression switching off, evidenced by $e_{i, j} = 0$, $e_{i,j}$ $\log_{2}(\mathrm{FC}_{i,j}) \gg 0$.

\begin{figure}
  \centering
  \includegraphics[width=.8\linewidth]{allocation_fc}
  \caption{
    $\log_{2}(\mathrm{FC})$ of enzyme usage reaction flux (Eq.\ \ref{eq:model-gecko-enzyme-usage}) in rounds of ablation, according to biomass component prioritised.
    %The cell changes how it allocates its proteome to enzymes when different components of its biomass are prioritised.
    Each column shows the component that remains in each round of pseudometabolite ablation (labels on top).
    Each row represents an enzyme, and rows are grouped by subsystem (labels on left).
    Colours represent $\log_{2}(\mathrm{FC})$ (defined in Eq.\ \ref{eq:model-foldchange}) showing how enzyme usage fluxes change in rounds of ablation: green shows an increase, while pink shows a decrease.
    Rows in which $|\log_{2}(\mathrm{FC})| < 11$ for all biomass components are excluded to restrict the number of reactions ($n = \num{3897}$) for visualisation.
  }
  \label{fig:model-ablate-enz-use}
\end{figure}

Figure~\ref{fig:model-ablate-enz-use} shows the subsystems and fold changes calculated as described above.
In particular, it shows:

\begin{enumerate}
  \item When any particular biomass component is prioritised, the cell de-allocates its proteome to most of its enzymes, as evidenced by $\log_{2}(\mathrm{FC}_{i,j}) < 0$ for most values of $i, j$.
        However, there are cases in which there are strong increases in allocation, as shown by $\log_{2}(\mathrm{FC}_{i,j}) > 0$.
        For example, when cofactors are prioritised, there are strong increases in biosynthesis of unsaturated fatty acids and in riboflavin metabolism.
        Additionally, there are strong increases in oxidative phosphorylation when carbohydrate, DNA, RNA, cofactor, or ion is prioritised.
  \item When the cell prioritises lipid biosynthesis, it exhibits the least change relative to the parallel (non-ablated case), compared to cases in which other biomass components are prioritised.
        The changes include increases of fluxes in the subsystems of fatty acid biosynthesis, glycerolipid metabolism, glycerophospholipid metabolism, inositol phosphate metabolism, steroid biosynthesis, and terpenoid backbone.
        Given that these subsystems are directly related to lipid metabolism, such changes are expected.
        In addition, during lipid biosynthesis, the model shows decreases of fluxes in oxidative phosphorylation, the TCA cycle, and in amino acid metabolism, with fluxes varying depending on the amino acid.
  \item When the cell prioritises protein biosynthesis, it shows small increases in fluxes associated with amino acid metabolism, tRNA metabolism, and oxidative phosphorylation.
        Such increases are expected as these led to production of substrates that are required for translation.
        Conversely, when other biomass components (carbohydrate, ion, DNA, RNA, and cofactor) are prioritised, there are decreases in fluxes in glycine, serine, and threonine metabolism.
  \item When the cell prioritises carbohydrate biosynthesis, there is an increase in fructose metabolism, mannose metabolism, N-glycan biosynthesis, and starch and sucrose metabolism, in contrast to the other biomass components.
        This is expected given that these reactions relate directly to pathways for synthesis of carbohydrates.
  \item When the cell prioritises RNA biosynthesis, there are increases in the purine metabolism and pyrimidine metabolism subsystems, along with a mixed picture in the pentose phosphate pathway.
        The increases are expected given the presence of purines and pyrimidines in RNA and the role of the pentose phosphate pathway in generating the precursors for these compounds.
        However, these subsystems show weak decreases in flux when DNA is prioritised, though the enzyme usage flux profile overall is similar to when RNA is prioritised.
\end{enumerate}

Overall, Fig.\ \ref{fig:model-ablate-enz-use} thus shows that in each round of ablation, when each biomass component is prioritised, the cell re-allocated its limited proteome to enzymes that have roles related to the synthesis of each biomass component.
These results validate ablation of biomass components as a method to simulate sequential synthesis of biomass components by the cell, as the changes of fluxes reflect cellular biochemistry.


\section{Estimating timescale of biosynthesis}
\label{sec:model-timescale}

% FIXME: Is this figure actually needed??  Or does it just confuse people??
\begin{figure}
  \centering
  \includegraphics[width=.7\linewidth]{ablation_example_adapted.png}
  \caption{
    Estimated synthesis times of biomass components, from ablation (blue bars) and from scaling doubling time by mass fraction (red bars).
    Bars under `all biomass' indicate (blue) $\Tseq$ (defined in Eq.\ \ref{eq:model-a}) and (red) the doubling time $t_{0}$ (defined in Eq.\ \ref{eq:model-doubling-time}).
    Other bars indicate (blue) $\Tiabl$ (defined in Eq.\ \ref{eq:model-ablated-time}) and (red) $\Tiprop$ (defined in Eq.\ \ref{eq:model-proportional-time}); $\Tpar$ is the greatest $\Tiprop$.
  }
  \label{fig:model-ablate-times}
\end{figure}


To evaluate whether sequential or parallel synthesis of biomass components offers a time advantage during cell growth, I estimated the timescale of biosynthesis for either resource allocation strategy (Fig.\ \ref{fig:model-ablate-times}).

Based on the objective function of the unmodified model, the doubling time is computed as follows:

\begin{equation}
  t_{0} = \frac{\ln 2}{\gro}
  \label{eq:model-doubling-time}
\end{equation}

where $t_{0}$ is the doubling time and $\gro$ is the growth rate, equivalent to the optimised flux of the biomass reaction.

Based on the growth rates computed in rounds of ablation (Fig.\ \ref{fig:model-ablate-fluxes}), the synthesis time of each biomass component is computed as follows:

\begin{equation}
  \Tiabl = f_{i} \cdot \frac{\ln 2}{\griabl}
  \label{eq:model-ablated-time}
\end{equation}

where
$i$ represents each of the biomass components (lipids, proteins, carbohydrates, DNA, RNA, cofactors, and ions),
$\Tiabl$ is the predicted time for synthesis of each biomass component,
$f_{i}$ is the mass fraction of each biomass component (shown in table~\ref{tab:model-biomfrac}), and
$\griabl$ is the optimal flux of the ablated biomass reaction.

Eq.\ \ref{eq:model-ablated-time} differs from Eq.\ \ref{eq:model-doubling-time} in that $f_{i}$ acts as a scaling factor that takes into account that if a biomass component accounts for a smaller fraction of dry cell mass, biomass synthesis should take less time.

For comparison, I computed estimates of the time for each biomass component, assuming that it is proportional to the mass fraction:

\begin{equation}
  \Tiprop = f_{i} \cdot t_{0}
  \label{eq:model-proportional-time}
\end{equation}

where $t_{0}$ is the doubling time found in Eq.\ \ref{eq:model-doubling-time}.

To determine whether sequential biosynthesis of biomass components or parallel biosynthesis of biomass components is advantageous, I devise a ratio $\ratioabl$ that represents the ratio between the total time predicted by ablation and the biomass component that is predicted to take the most time.
This is defined:

\begin{equation}
  \ratioabl = \frac{\Tseq}{\Tpar}
  \label{eq:model-ratio-simplified}
\end{equation}

where $\Tseq$ represents the predicted growth time assuming sequential biosynthesis and $\Tpar$ represents the limiting biomass synthesis time assuming parallel biosynthesis.
These quantities are defined:

\begin{equation}
  \Tseq = \sum_{i} \Tiabl = \sum f_{i} \cdot \frac{\ln 2}{\griabl}
  \label{eq:model-a}
\end{equation}

% \begin{equation}
%   \begin{aligned}
%     \Tpar \coloneqq \argmax_{i} \Tiprop = \Tabl{protein} = \biomfrac{protein} \cdot \frac{\ln 2}{\gro},\\
%     \because \argmax_{i} f_{i} = \biomfrac{protein}
%   \end{aligned}
%   \label{eq:model-b}
% \end{equation}

\begin{equation}
  \Tpar = \argmax_{i} \Tiprop
  \label{eq:model-b}
\end{equation}

As $\biomfrac{protein} = 0.525$, $\Tpar = \Tprop{protein}$.

Therefore,

\begin{equation}
  \begin{aligned}
    \ratioabl &= \frac{\Tseq}{\Tpar} \\
    & = \left( \sum_i \frac{f_i}{\griabl} \right) \cdot \frac{\gro}{\biomfrac{protein}} \\
    & = \left( \frac{\biomfrac{lipid}}{\grabl{lipid}} + \frac{\biomfrac{protein}}{\grabl{protein}} + \ldots + \frac{\biomfrac{ion}}{\grabl{ion}} \right) \cdot \frac{\gro}{\biomfrac{protein}}
    \end{aligned}
  \label{eq:model-ratio}
\end{equation}

The expression in Eq.\ \ref{eq:model-ratio} means that the definition of the $\ratioabl$ ratio does not reduce to a trivial expression and depends on the $\gro$ and the $\griabl$ values, which are all independent of each other.
%This relationship between $\ratioabl$ and $\gro$ and $\griabl$ is explored in section~\ref{sec:model-pool}.
A $\ratioabl < 1$ means that synthesising biomass components in sequence saves more time, favouring sequential biosynthesis.
Conversely, $\ratioabl > 1$ indicates that parallel synthesis of biomass components is favoured as synthesising biomass components in sequence does not save time.


\begin{figure}
  \centering
  \begin{subfigure}[htpb]{0.45\textwidth}
   \centering
   \includegraphics[width=\textwidth]{ablation_by4741}
   \caption{
   }
   \label{fig:model-ablation-by4741}
  \end{subfigure}
  \begin{subfigure}[htpb]{0.45\textwidth}
   \centering
   \includegraphics[width=\textwidth]{ablation_zwf1}
   \caption{
   }
   \label{fig:model-ablation-zwf1}
  \end{subfigure}
  \begin{subfigure}[htpb]{0.45\textwidth}
   \centering
   \includegraphics[width=\textwidth]{ablation_tsa2}
   \caption{
   }
   \label{fig:model-ablation-tsa2}
  \end{subfigure}
  \caption{
    Estimated synthesis times of biomass components, from ablation (blue bars) and from scaling doubling time by mass fraction (red bars), for \textbf{(\ref{fig:model-ablation-by4741})} BY4741, \textbf{(\ref{fig:model-ablation-zwf1})} zwf$\Delta$ in the BY4741 background, and \textbf{(\ref{fig:model-ablation-tsa2})} tsa2$\Delta$ in the BY4742 background.
    %The ecYeast8 model does not include reactions that correspond to \textit{TSA1}.
    % For BY4741-background strains, supplements are simulated by allowing uptake of histidine, leucine, tryptophan, methionine and uracil.
    % The same applied to BY4742-background strains, but lysine uptake replaces methionine uptake.
  }
  \label{fig:model-ablation-strains}
\end{figure}

% In figure~\ref{fig:model-ablate-times}, the $\ratioabl = 0.70$.
To confirm that the advantage of the sequential biosynthesis strategy is retained in other strains, I extended the computation $\ratioabl$ and related quantities to models in which genes were deleted.
In accordance with chapter 3, the BY4741 zwf1$\Delta$ and BY4742 tsa1$\Delta$ tsa2$\Delta$ strains were simulated.
The deletions were made by restricting reaction fluxes that are associated with the deleted genes to zero.
For BY4741-background strains, supplements are simulated by allowing uptake of histidine, leucine, tryptophan, methionine and uracil.
The same applied to BY4742-background strains, but lysine uptake replaces methionine uptake.
Fig.\ \ref{fig:model-ablation-strains} shows that $\ratioabl < 1$ still holds for auxotrophs and deletion strains, suggesting that the sequential biosynthesis strategy remains advantageous.
Assuming that temporal scheduling of the synthesis of biomass components during growth explains the timing of biosynthetic events in the yeast metabolic cycle, these observations supports results in chapter 3 which shows that auxotrophs and deletion strains have YMCs.


\section{Effect of restricting the enzyme pool}
\label{sec:model-pool}

The yeast cell has a finite enzyme-available proteome pool, so it must decide which enzymes to allocate the greatest proportions of the pool to, for enzyme synthesis.
In order to study this effect using the ecYeast8 model, a constraint on the proteome pool must be imposed, and I imposed this constraint by varying the enzyme-available proteome pool defined by the enzyme pool pseudoreaction (Eq.\ \ref{eq:model-gecko-enzyme-pool-flux}).
To vary the enzyme-available proteome pool in the ecYeast8 model,
I vary the value of the upper limit of the flux $\epool$ of the enzyme pool pseudoreaction (equation~\ref{eq:model-gecko-enzyme-pool}) in order to change the flux available for enzyme usage pseudoreactions (equation~\ref{eq:model-gecko-enzyme-usage}).
With a smaller $\epool$, the sum of fluxes of enzyme usage pseudoreactions must decrease, and the model must decide which enzyme usage pseudoreactions to allocate a higher flux to, modelling the biological response to a restricted enzyme pool.

\begin{figure}
  \centering
  \begin{subfigure}[htpb]{0.45\textwidth}
   \centering
   \includegraphics[width=\textwidth]{epool_ec_ratio_shrinkyaxis}
   \caption{
   }
   \label{fig:model-pool-ratio}
  \end{subfigure}%
  \begin{subfigure}[htpb]{0.45\textwidth}
   \centering
   \includegraphics[width=\textwidth]{epool_ec_ratio_20_shrinkyaxis}
   \caption{
   }
   \label{fig:model-pool-ratio-20}
  \end{subfigure}

  \begin{subfigure}[htpb]{0.45\textwidth}
   \centering
   \includegraphics[width=\textwidth]{epool_ec_gr}
   \caption{
   }
   \label{fig:model-pool-growthrate}
  \end{subfigure}%
  \begin{subfigure}[htpb]{0.45\textwidth}
   \centering
   \includegraphics[width=\textwidth]{epool_ec_gr_20}
   \caption{
   }
   \label{fig:model-pool-growthrate-20}
  \end{subfigure}

  \begin{subfigure}[htpb]{0.45\textwidth}
   \centering
   \includegraphics[width=\textwidth]{epool_ec_components}
   \caption{
   }
   \label{fig:model-pool-ablated}
  \end{subfigure}%
  \begin{subfigure}[htpb]{0.45\textwidth}
   \centering
   \includegraphics[width=\textwidth]{epool_ec_components_20}
   \caption{
   }
   \label{fig:model-pool-ablated-20}
  \end{subfigure}

  \caption{
    Effect of the size of the proteome pool available for enzymes ($\epool^{\prime}$) on \textbf{(\ref{fig:model-pool-ratio}, \ref{fig:model-pool-ratio-20})} $\ratioabl$, \textbf{(\ref{fig:model-pool-growthrate}, \ref{fig:model-pool-growthrate-20})} the growth rate, and \textbf{(\ref{fig:model-pool-ablated}, \ref{fig:model-pool-ablated-20})} the optimal flux of the ablated biomass reaction $\griabl$ in each round of ablation.
    \textbf{(\ref{fig:model-pool-ratio})}, \textbf{(\ref{fig:model-pool-growthrate})}, and \textbf{(\ref{fig:model-pool-ablated})} show $\epool^{\prime}/\epool \leq 2$, while
    \textbf{(\ref{fig:model-pool-ratio-20})}, \textbf{(\ref{fig:model-pool-growthrate-20})}, and \textbf{(\ref{fig:model-pool-ablated-20})} show $\epool^{\prime}/\epool \leq 20$, while
  }
  \label{fig:model-pool}
\end{figure}

Fig.\ \ref{fig:model-pool} shows that constraining the proteome pool available for enzymes leads to a greater advantage of sequential biosynthesis of biomass components over parallel biosynthesis.
This is evidenced by a decreasing $\ratioabl$ ratio if $\epool^{\prime}$ decreases, within the range of $\epool^{\prime}$ that gives realistic growth rates (Fig.\ \ref{fig:model-pool-ratio}).
Concurrently, as $\epool^{\prime}$ decreases, the wild type growth rate ($\gro$) decreases linearly to zero (Fig.\ \ref{fig:model-pool-growthrate}) and ablated growth rates ($\griabl$) decrease in linear segments independently of each other and of the growth rate (Fig.\ \ref{fig:model-pool-growthrate}).
These observations can be explained by considering Eq.\ \ref{eq:model-ratio} and modelling changes in $\gro$ and $\griabl$ with respect to $\epool^{\prime}/\epool$ as linear equations (Appendix \ref{append:model-pool}).


\section{Effect of carbon and nitrogen sources}
\label{sec:model-exchange}

To explore whether sequential synthesis of biomass components remains advantageous across nutrient conditions, I investigated how changes in the concentrations of nitrogen and carbon sources affect the resource allocation strategy in yeast.
The nitrogen source of interest was ammonium as it is the form of nitrogen present in minimal growth media.
The carbon sources of interest included glucose and pyruvate.
Glucose was studied as it is the preferred carbon source for budding yeast, while pyruvate served as an example of a non-fermentable carbon source to investigate the effect of significant changes to central carbon metabolism on resource allocation strategies.

% FIXME: Be consistent in referring to exchange rate/saturation.  Come up with some phrase to use.
\subsection{Saturation of exchange reactions}
\label{subsec:model-saturation}

Investigating the effect of carbon and nitrogen sources requires finding ranges of concentrations of each source to be used in an FBA problem that leads to biologically informative results.
The saturation point of the nutrient, defined as the concentration of the nutrient at which the growth rate reaches its maximum, thus serves as a biologically relevant reference point.
Although genome-scale metabolic models have nutrient exchange reactions that simulate the presence or absence of nutrients, the saturation point concentrations from experimental studies cannot be directly used as flux bounds for these reactions because FBA does not account for substrate concentrations.
Instead, I constrain the flux of exchange reactions in such a way that the optimised growth rate matches experimental observations, as was performed in previous FBA-based studies \parencite{elsemmanWholecellModelingYeast2022}.
% TODO: Find other studies that did the same and cite them here.


\begin{figure}
  \centering
  \begin{subfigure}[t]{0.45\textwidth}
  \centering
    \includegraphics[width=\linewidth]{saturation_glc}
    \caption{
      % Effect of glucose exchange on growth rate, with ammonium exchange flux unrestricted.
      % Growth rate saturation is at \SI{8.69}{\mmolgdwh}.
    }
    \label{fig:model-saturation-glucose}
  \end{subfigure}%
  \begin{subfigure}[t]{0.45\textwidth}
  \centering
    \includegraphics[width=\linewidth]{saturation_pyr}
    \caption{
      % Effect of pyruvate exchange on growth rate, with ammonium exchange flux unrestricted.
      % Growth rate saturation is at \SI{4.44}{\mmolgdwh}.
    }
    \label{fig:model-saturation-pyruvate}
  \end{subfigure}

  \begin{subfigure}[t]{0.45\textwidth}
  \centering
    \includegraphics[width=\linewidth]{saturation_amm}
    \caption{
      % Effect of ammonium exchange on growth rate, with exchanges of carbon sources set to growth rate saturation based on figures~\ref{fig:model-saturation-glucose} and ~\ref{fig:model-saturation-pyruvate}.
      % Growth rate saturation is at \SI{1.48}{\mmolgdwh} in glucose, and
      % at \SI{1.00}{\mmolgdwh} in pyruvate.
    }
    \label{fig:model-saturation-ammonium}
  \end{subfigure}

  \caption{
    Effect of glucose \textbf{(\ref{fig:model-saturation-glucose})}, pyruvate \textbf{(\ref{fig:model-saturation-pyruvate})}, and ammonium \textbf{(\ref{fig:model-saturation-ammonium})} exchange reactions on growth rate.
    The saturation point of exchange reactions is defined as the flux of an exchange reaction above which the objective function reaches its maximum.
  }
  \label{fig:model-saturation}
\end{figure}

To model the effect of nutrient concentrations on growth rate, I created saturation curves that show how the effect of nutrient exchange flux on the objective function.
These curves were created by varying the upper bounds of nutrient exchange reactions and optimising the model at each upper bound value.
Figs.\ \ref{fig:model-saturation-glucose} and \ref{fig:model-saturation-pyruvate} show that the saturation curve of glucose exchange and of pyruvate exchange exhibited different shapes.
In addition, the maximum growth rate on glucose (\SI{0.38}{\hour^{-1}}) was greater than the maximum growth rate on pyruvate (\SI{0.25}{\hour^{-1}}).
Furthermore, the growth rate saturation point for ammonium was determined by the maximal growth rate on the carbon source (Fig.\ \ref{fig:model-saturation-ammonium}).

These growth saturation curves agree with similar studies.
Specifically, the growth saturation curve for glucose (Fig.\ \ref{fig:model-saturation-glucose}) is similar to that simulated by \textcite{elsemmanWholecellModelingYeast2022} using another derivative of the Yeast8 model.
The maximum growth rate on glucose agrees with \textcite{domenzainReconstructionCatalogueGenomescale2022}, which used GECKO 2 to create the ecYeast7 model to predict maximum growth rates on various carbon sources.
\textcite{domenzainReconstructionCatalogueGenomescale2022} did not simulate growth on pyruvate, but a lower maximum growth rate on pyruvate is consistent with my experimental observations.


\subsection{Effect of carbon and nitrogen sources on biomass synthesis strategies}
\label{subsec:model-grid}

\subsubsection{Glucose and ammonium}
\label{subsec:model-grid-glucose}

\begin{figure}
  \centering
  \begin{subfigure}[t]{0.45\textwidth}
  \centering
    \includegraphics[width=\linewidth]{ec_grid_glc_amm_ratio}
    \caption{
    }
    \label{fig:model-grid-glc-ratio}
  \end{subfigure}%
  \begin{subfigure}[t]{0.45\textwidth}
  \centering
    \includegraphics[width=\linewidth]{ec_grid_glc_amm_gr}
    \caption{
    }
    \label{fig:model-grid-glc-growthrate}
  \end{subfigure}

  \begin{subfigure}[t]{0.45\textwidth}
  \centering
    \includegraphics[width=\linewidth]{ec_grid_glc_amm_carb}
    \caption{
    }
    \label{fig:model-grid-glc-carb}
  \end{subfigure}%
  \begin{subfigure}[t]{0.45\textwidth}
  \centering
    \includegraphics[width=\linewidth]{ec_grid_glc_amm_prot}
    \caption{
    }
    \label{fig:model-grid-glc-prot}
  \end{subfigure}

  \begin{subfigure}[t]{0.45\textwidth}
  \centering
    \includegraphics[width=\linewidth]{ec_grid_glc_amm_carb_to_prot}
    \caption{
    }
    \label{fig:model-grid-glc-carb-to-prot}
  \end{subfigure}
  \caption{
    Effect of glucose and ammonium exchange rates on \textbf{(\ref{fig:model-grid-glc-ratio})} $\ratioabl$, \textbf{(\ref{fig:model-grid-glc-growthrate})} growth rate based on unmodified biomass reaction, $\gro$, \textbf{(\ref{fig:model-grid-glc-carb})} predicted carbohydrate synthesis time, $\Tabl{carbohydrate}$ \textbf{(\ref{fig:model-grid-glc-prot})} predicted protein synthesis time, $\Tabl{protein}$, and \textbf{(\ref{fig:model-grid-glc-carb-to-prot})} ratio of carbohydrate synthesis time to protein synthesis time, $\Tabl{carbohydrate}/\Tabl{protein}$.
    Exchange rates are expressed in percentages of growth saturation from Fig.\ \ref{fig:model-saturation}, with black straight lines indicate 100\% of saturation.
    Contours show regions in which $\ratioabl > 1$.
    Arrows overlaid on heatmaps indicate sensitivity of the quantity as a vector $(s_{\mathrm{glc}}, s_{\mathrm{amm}})$ (Eq.\ \ref{eq:model-susceptibility}).
  }
  \label{fig:model-grid-glc}
\end{figure}
% To visualise the extent to which each exchange rate limits each quantity of interest, I show the sensitivity values as arrows (Figs.\ \ref{fig:model-grid-glc} and~\ref{fig:model-grid-pyr}) with $s_{\mathrm{glc}}$ and $s_{\mathrm{amm}}$ as two components of the vector that defines these arrows.
% Thus, horizontal arrows indicate a strong sensitivity to glucose and vertical arrows indicate a strong sensitivity to ammonium, while the arrows point towards increasing values of the quantity of interest.

To assess the effect of glucose and ammonium concentration on biomass synthesis strategies, I ablated components in the biomass reaction in different nutrient conditions set by glucose and ammonium exchange fluxes
to obtain $\ratioabl$, $\gro$, $\Tabl{carbohydrate}$, and $\Tabl{protein}$ (Fig.\ \ref{fig:model-grid-glc}).
Additionally, to determine whether the carbon or nitrogen source is limiting for each of the quantities calculated, I defined the sensitivity at each nutrient condition $(\exchrate{glc}, \exchrate{amm})$, with respect to each axis $i$ as:

\begin{equation}
  s_{i}(\exchrate{glc}, \exchrate{amm}) = \frac{R_{i}}{y(\exchrate{glc}, \exchrate{amm})} \cdot \pdif{y(\exchrate{glc}, \exchrate{amm})}{R_{i}}
  \label{eq:model-susceptibility}
\end{equation}

where
$i$ indicates glucose (glc) or ammonium (amm),
$R_{i}$ indicates the exchange rate, glucose ($\exchrate{glc}$) or ammonium ($\exchrate{amm}$),
an $(\exchrate{glc}, \exchrate{amm})$ pair defines a nutrient condition, and
$y(\exchrate{glc}, \exchrate{amm})$ represents the quantity of interest at each nutrient condition.

At a specific nutrient condition $(\exchrate{glc}, \exchrate{amm})$, if $s_{\mathrm{glc}}(\exchrate{glc}, \exchrate{amm}) > s_{\mathrm{amm}}(\exchrate{glc}, \exchrate{amm})$ for a quantity of interest $y$, then it means that the quantity is more sensitive to glucose exchange at this condition.
The reverse is true if $s_{\mathrm{glc}}(\exchrate{glc}, \exchrate{amm}) < s_{\mathrm{amm}}(\exchrate{glc}, \exchrate{amm})$.

Fig.\ \ref{fig:model-grid-glc-growthrate} divides the nutrient conditions into three regions according to the nutrients that limit growth rate as determined by sensitivity values: a glucose-limiting region, an ammonium-limiting region, and a region whether neither glucose nor ammonium is limiting.
Fig.\ \ref{fig:model-grid-glc-ratio} shows that parallel biosynthesis is advantageous ($\ratioabl > 1$) in two circumstances: when both glucose and ammonium limit the growth rate, and when ammonium exchange is at saturation while glucose exchange is greater than saturation.
In contrast, when the growth rate is near its maximum, where neither glucose nor ammonium is limiting, when glucose or ammonium exchange increases, sequential biosynthesis becomes more advantageous ($\ratioabl < 1$).

To determine whether changes in the synthesis time of biomass components are responsible for switching of scheduling strategies between sequential and parallel biosynthesis, I investigated how $\exchrate{glc}$ and $\exchrate{amm}$ affected $\frac{\Tabl{carbohydrate}}{\Tabl{protein}}$.
Synthesis times $\Tiabl$ were computed to assess whether the ratio between the synthesis times differed in different conditions.
The ratio $\frac{\Tabl{carbohydrate}}{\Tabl{protein}}$ serves as the principal measure to quantify how $\Tiabl$ values change as nutrient conditions change.
This ratio is informative because:
\begin{enumerate}
  \item Of all biomass components, predicted synthesis times of these biomass components varied the most as the glucose and ammonium exchange rates were varied.
  \item Each accounts for a large proportion of biomass: protein accounts for 52.5\% and carbohydrate accounts for 36.4\% (table~\ref{tab:model-biomfrac}).
  \item Carbohydrate synthesis has a clear biochemical relationship with the level of a carbon source.
        Additionally, protein synthesis has a clear biochemical relationship with the level of ammonium as amino acids contain amino groups.
\end{enumerate}

% Based on equation~\ref{eq:model-ablated-time}, consider:

% \begin{equation}
%   \frac{\Tabl{carbohydrate}}{\Tabl{protein}} = \frac{\biomfrac{carbohydrate}}{\biomfrac{protein}} \cdot \frac{\grabl{protein}}{\grabl{carbohydrate}}
%   \label{eq:model-carb-prot-ratio}
% \end{equation}

Although there is no obvious relationship between $\frac{\Tabl{carbohydrate}}{\Tabl{protein}}$ and $\ratioabl$ values (Fig.\ \ref{fig:model-grid-glc-carb-to-prot}),
the interplay between how $\exchrate{glc}$ and $\exchrate{pyr}$ affect $\Tabl{carbohydrate}$ and $\Tabl{protein}$ explains the landscape of $\frac{\Tabl{carbohydrate}}{\Tabl{protein}}$ values.
Fig.\ \ref{fig:model-grid-glc-carb-to-prot} shows that in the glucose-limiting region, $\frac{\Tabl{carbohydrate}}{\Tabl{protein}}$ is at a constant high value because, as $\exchrate{glc}$ increases in this region, both $\Tabl{carbohydrate}$ and $\Tabl{protein}$ decrease.
However, in the ammonium-limited region, $\frac{\Tabl{carbohydrate}}{\Tabl{protein}}$ increases as $\exchrate{amm}$ increases because $\Tabl{protein}$ decreases while $\Tabl{carbohydrate}$ remains constant.

Consider the trajectory of $(\exchrate{glc}, \exchrate{amm})$ values from $(0, 0)$ to $(\exchrate{glc, sat}, \exchrate{amm, sat})$ that links conditions with the high $\ratioabl$ values, where $\exchrate{glc, sat}$ and $\exchrate{amm, sat}$ are saturation values of glucose and ammonium values respectively.
This trajectory includes conditions in which both glucose and ammonium are limiting, and as the trajectory is followed, both $\Tabl{carbohydrate}$ and $\Tabl{protein}$ decrease, explaining why $\frac{\Tabl{carbohydrate}}{\Tabl{protein}}$ remains roughly at the same value.

The different effects of the exchange rates on $\Tabl{carbohydrate}$ and on $\Tabl{protein}$ as discussed above have a biological relevance.
Fig.\ \ref{fig:model-grid-glc-carb} shows that glucose exchange limits carbohydrate synthesis time in all nutrient conditions (Fig.\ \ref{fig:model-grid-glc-carb}), and this observation can be explained by glucose being the sole carbon source for the cell.
In contrast, Fig.\ \ref{fig:model-grid-glc-prot} shows that ammonium exchange limits protein synthesis time for most nutrient conditions, except for conditions in which glucose exchange is very low.
This observation can be explained by the fact that both carbon and nitrogen sources contribute to protein biosynthesis.
% This sounds like a pain point to be emphasised.
In sum, the relationship between carbon and nitrogen source uptake and the synthesis of carbohydrates and proteins determines how scheduling strategies change according to nutrient conditions.


\subsubsection{Pyruvate and ammonium}
\label{subsec:model-grid-pyruvate}

\begin{figure}
  \centering
  \begin{subfigure}[t]{0.45\textwidth}
  \centering
    \includegraphics[width=\linewidth]{ec_grid_pyr_amm_ratio}
    \caption{
    }
    \label{fig:model-grid-pyr-ratio}
  \end{subfigure}%
  \begin{subfigure}[t]{0.45\textwidth}
  \centering
    \includegraphics[width=\linewidth]{ec_grid_pyr_amm_gr}
    \caption{
    }
    \label{fig:model-grid-pyr-growthrate}
  \end{subfigure}

  \begin{subfigure}[t]{0.45\textwidth}
  \centering
    \includegraphics[width=\linewidth]{ec_grid_pyr_amm_carb}
    \caption{
    }
    \label{fig:model-grid-pyr-carb}
  \end{subfigure}%
  \begin{subfigure}[t]{0.45\textwidth}
  \centering
    \includegraphics[width=\linewidth]{ec_grid_pyr_amm_prot}
    \caption{
    }
    \label{fig:model-grid-pyr-prot}
  \end{subfigure}

  \begin{subfigure}[t]{0.45\textwidth}
  \centering
    \includegraphics[width=\linewidth]{ec_grid_pyr_amm_carb_to_prot}
    \caption{
    }
    \label{fig:model-grid-pyr-carb-to-prot}
  \end{subfigure}
  \caption{
    Effect of pyruvate and ammonium exchange rates on \textbf{(\ref{fig:model-grid-pyr-ratio})} $\ratioabl$, \textbf{(\ref{fig:model-grid-pyr-growthrate})} growth rate based on unmodified biomass reaction, $\gro$, \textbf{(\ref{fig:model-grid-pyr-carb})} predicted carbohydrate synthesis time, $\Tabl{carbohydrate}$ \textbf{(\ref{fig:model-grid-pyr-prot})} predicted protein synthesis time, $\Tabl{protein}$, and \textbf{(\ref{fig:model-grid-pyr-carb-to-prot})} ratio of carbohydrate synthesis time to protein synthesis time, $\Tabl{carbohydrate}/\Tabl{protein}$.
    Exchange rates are expressed in percentages of growth saturation from Fig.\ \ref{fig:model-saturation}, with black straight lines indicate 100\% of saturation.
    Contours show regions in which $\ratioabl > 1$.
    Arrows overlaid on heatmaps indicate sensitivity of the quantity as a vector $(s_{\mathrm{pyr}}, s_{\mathrm{amm}})$ (Eq.\ \ref{eq:model-susceptibility}).
  }
  \label{fig:model-grid-pyr}
\end{figure}

To investigate the effect of a non-fermentable carbon source on biomass synthesis strategies, I repeated the investigation using pyruvate as the carbon source, ablating components in the biomass reaction in different nutrient conditions.%, and adjusted the saturation exchange rates accordingly: pyruvate saturation being at \SI{4.44}{\mmolgdwh} and ammonium saturation being at \SI{1.00}{\mmolgdwh} (figure~\ref{fig:model-grid-pyr}).

Results show that the saturation behaviour of pyruvate controlled the effect of exchange rates on scheduling strategies.
Fig.\ \ref{fig:model-grid-pyr} shows when pyruvate was the carbon source, the region of carbon source limitation was larger than when glucose was the carbon source.
This observation reflected how the pyruvate saturation curve differed from the glucose saturation curve (Fig.\ \ref{fig:model-saturation}).
Namely, as pyruvate exchange increased below saturation, the growth rate increased linearly: the growth rate reached half its maximum value when pyruvate exchange was half of its saturation value.
In contrast, as glucose exchange increased below saturation, the growth rate did not increase linearly: the growth rate reached half its maximum value when glucose exchange was well below half of its saturation value.

Fig.\ \ref{fig:model-grid-pyr} shows that the relationship between exchange rates and the quantities $\ratioabl$, $\gro$, $\Tabl{carbohydrate}$, and $\Tabl{protein}$ was similar to the glucose-ammonium case.
In other words, when pyruvate was the carbon source, the relationship between carbon and nitrogen source uptake and the synthesis of carbohydrates and proteins still determined how scheduling strategies changed according to nutrient conditions.
Parallel biosynthesis was advantageous in two conditions: when both pyruvate and ammonium limited the growth rate, and when ammonium exchange was at saturation while pyruvate exchange was greater than saturation (Fig.\ \ref{fig:model-grid-pyr-ratio}).
In contrast, sequential biosynthesis was advantageous when neither pyruvate nor ammonium was limiting.
$\frac{\Tabl{carbohydrate}}{\Tabl{protein}}$ remained at high values when pyruvate is limiting or when pyruvate and ammonium were both limiting (Fig.\ \ref{fig:model-grid-pyr-carb-to-prot}).
Furthermore, the behaviour of $\frac{\Tabl{carbohydrate}}{\Tabl{protein}}$ as nutrient conditions varied could also be explained by how pyruvate exchange limited carbohydrate synthesis time (Fig.\ \ref{fig:model-grid-pyr-carb}) and by how both pyruvate and ammonium exchange limited protein synthesis time (Fig.\ \ref{fig:model-grid-pyr-prot}).


\subsection{Relationship between proteome allocation and resource allocation strategy}
\label{subsec:model-rank}

\subsubsection{Glucose and ammonium}
\label{subsec:model-rank-glucose}

\begin{figure}
  \centering
  \begin{subfigure}[t]{0.45\textwidth}
  \centering
    \includegraphics[width=\linewidth]{CompareEnzUse_glc01p69_pyrUnres_amm01p05_1.pdf}
    \caption{
    }
    \label{fig:model-rank-glc-highratio-rank}
  \end{subfigure}%
  \begin{subfigure}[t]{0.45\textwidth}
  \centering
    \includegraphics[width=\linewidth]{CompareEnzUse_glc16p89_pyrUnres_ammUnres_1.pdf}
    \caption{
    }
    \label{fig:model-rank-glc-lowratio-rank}
  \end{subfigure}%

  \caption{
    Changes in enzyme usage reaction flux in rounds of ablation, in  \textbf{(\ref{fig:model-rank-glc-highratio-rank})} a high $\ratioabl$ condition ($\exchrate{glc}$ = \SI{1.69}{\mmolgdwh}, $\exchrate{amm}$ = \SI{1.05}{\mmolgdwh}), and \textbf{(\ref{fig:model-rank-glc-lowratio-rank})} a low $\ratioabl$ condition ($\exchrate{glc}$ = \SI{16.89}{\mmolgdwh}).
    Columns show the biomass component prioritised.
    In each column, rows represent enzyme usage reactions, arranged in descending order of flux.
    Colours identify the reactions, with white indicating reactions that carry zero flux in the parallel case.
  }
  \label{fig:model-rank-glc-rank}
\end{figure}

\begin{figure}
  \centering
  \begin{subfigure}[t]{0.45\textwidth}
  \centering
    \includegraphics[width=\linewidth]{CompareEnzUse_glc01p69_pyrUnres_amm01p05_2.pdf}
    \caption{
    }
    \label{fig:model-rank-glc-highratio-kendall}
  \end{subfigure}%
  \begin{subfigure}[t]{0.45\textwidth}
  \centering
    \includegraphics[width=\linewidth]{CompareEnzUse_glc16p89_pyrUnres_ammUnres_2.pdf}
    \caption{
    }
    \label{fig:model-rank-glc-lowratio-kendall}
  \end{subfigure}

  \caption{
    Pairwise Kendall's $\tau$-b rank correlation coefficient \parencite{kendallTREATMENTTIESRANKING1945} for each pair enzyme usage flux profiles represented in columns of Fig.\ \ref{fig:model-rank-glc-rank} of columns in the top panels to quantify the similarity between each case.
    The computations were performed for \textbf{(\ref{fig:model-rank-glc-highratio-kendall})} a high $\ratioabl$ condition ($\exchrate{glc}$ = \SI{1.69}{\mmolgdwh}, $\exchrate{amm}$ = \SI{1.05}{\mmolgdwh}), and \textbf{(\ref{fig:model-rank-glc-lowratio-kendall})} a low $\ratioabl$ condition ($\exchrate{glc}$ = \SI{16.89}{\mmolgdwh}).
  }
  \label{fig:model-rank-glc-kendall}
\end{figure}

The investigation of the effect of carbon and nitrogen sources on scheduling strategies led to a new hypothesis: the cell favours parallel biosynthesis of biomass components when the carbon source and the nitrogen source are both limiting, especially if the associated biomass components share metabolic pathways and if the conditions dictate similar levels of enzymes.
This hypothesis arose from the observation that, in both the glucose-ammonium and pyruvate-ammonium conditions, the highest $\ratioabl$ occurred at the boundary at which both the carbon source and the nitrogen source were limiting.

To evaluate the hypothesis, I compared the enzyme usage reaction flux vectors in rounds of ablation between a high $\ratioabl$ condition and a low $\ratioabl$ condition created by combinations of glucose and nitrogen exchange rates.
Enzyme usage reaction flux vectors show how the cell allocates its finite proteome to enzymes when each biomass component is prioritised (section~\ref{sec:model-timescale}).
To test whether similarities in proteome allocation to enzymes explain parallel biosynthesis, Fig.\ \ref{fig:model-rank-glc-highratio-rank} shows that when both glucose and ammonium are limiting, the allocation patterns are similar across rounds of ablation, thus explaining the advantage of parallel biosynthesis of biomass components.
The similarity is quantified in Fig.\ \ref{fig:model-rank-glc-highratio-kendall}, which shows that parallel biosynthesis exhibits relatively similar proteome allocation with the synthesis of individual biomass components.
In contrast, Fig.\ \ref{fig:model-rank-glc-lowratio-rank} shows that when neither glucose nor ammonium limits growth, biomass components can be divided into two groups based on enzyme usage reaction flux vectors.
These groups suggest an explanation for sequential biosynthesis: the cell decides to synthesise lipids and proteins together in one stage of its growth cycle, while synthesising carbohydrates, DNA, RNA, cofactors, and ions together in another stage.
The bias towards sequential biosynthesis is further emphasised by Fig.\ \ref{fig:model-rank-glc-lowratio-kendall}, which shows that parallel biosynthesis exhibits different proteome allocation compared to the synthesis of individual biomass components.


\subsubsection{Pyruvate and ammonium}
\label{subsec:model-rank-pyruvate}

\begin{figure}
  \centering
  \begin{subfigure}[t]{0.45\textwidth}
  \centering
    \includegraphics[width=\linewidth]{CompareEnzUse_glc00p00_pyr03p73_amm00p90_1.pdf}
    \caption{
    }
    \label{fig:model-rank-pyr-highratio-rank}
  \end{subfigure}%
  \begin{subfigure}[t]{0.45\textwidth}
  \centering
    \includegraphics[width=\linewidth]{CompareEnzUse_glc00p00_pyr08p89_ammUnres_1.pdf}
    \caption{
    }
    \label{fig:model-rank-pyr-lowratio-rank}
  \end{subfigure}

  \caption{
    Changes in enzyme usage reaction flux in rounds of ablation, in  \textbf{(\ref{fig:model-rank-pyr-highratio-rank})} a high $\ratioabl$ condition ($\exchrate{pyr}$ = \SI{3.73}{\mmolgdwh}, $\exchrate{amm}$ = \SI{0.90}{\mmolgdwh}), and \textbf{(\ref{fig:model-rank-pyr-lowratio-rank})} a low $\ratioabl$ condition ($\exchrate{pyr}$ = \SI{8.89}{\mmolgdwh}).
    Columns show the biomass component prioritised.
    In each column, rows represent enzyme usage reactions, arranged in descending order of flux.
    Colours identify the reactions, with white indicating reactions that carry zero flux in the parallel case.
  }
  \label{fig:model-rank-pyr-rank}
\end{figure}

\begin{figure}
  \centering
  \begin{subfigure}[t]{0.45\textwidth}
  \centering
    \includegraphics[width=\linewidth]{CompareEnzUse_glc00p00_pyr03p73_amm00p90_2.pdf}
    \caption{
    }
    \label{fig:model-rank-pyr-highratio-kendall}
  \end{subfigure}%
  \begin{subfigure}[t]{0.45\textwidth}
  \centering
    \includegraphics[width=\linewidth]{CompareEnzUse_glc00p00_pyr08p89_ammUnres_2.pdf}
    \caption{
    }
    \label{fig:model-rank-pyr-lowratio-kendall}
  \end{subfigure}

  \caption{
    Pairwise Kendall's $\tau$-b rank correlation coefficient \parencite{kendallTREATMENTTIESRANKING1945} for each pair enzyme usage flux profiles represented in columns of Fig.\ \ref{fig:model-rank-pyr-rank} of columns in the top panels to quantify the similarity between each case.
    The computations were performed for \textbf{(\ref{fig:model-rank-pyr-highratio-kendall})} a high $\ratioabl$ condition ($\exchrate{pyr}$ = \SI{3.73}{\mmolgdwh}, $\exchrate{amm}$ = \SI{0.90}{\mmolgdwh}), and \textbf{(\ref{fig:model-rank-pyr-lowratio-kendall})} a low $\ratioabl$ condition ($\exchrate{pyr}$ = \SI{8.89}{\mmolgdwh}).
  }
  \label{fig:model-rank-pyr-kendall}
\end{figure}

However, Fig.\ \ref{fig:model-rank-pyr-rank} suggests that the contrast between each biomass component is lessened when pyruvate is the carbon source.

To evaluate whether the hypothesis that parallel biosynthesis is favoured when biomass components share metabolic pathways in a different carbon source, I extended the comparison of enzyme usage reaction flux vectors in rounds of ablation to nutrient conditions created by pyruvate and ammonium exchange rates.

Similar to the glucose-ammonium investigation, in the high $\ratioabl$ condition where pyruvate and ammonium are limiting, Figs.\ \ref{fig:model-rank-pyr-highratio-rank} and \ref{fig:model-rank-pyr-highratio-kendall} show that proteome allocation were similar across rounds of ablation, supporting parallel biosynthesis as a scheduling strategy.
However, the results for the low $\ratioabl$ case suggest that grouping of biomass components explain sequential biosynthesis for only very low $\ratioabl$.
Fig.\ \ref{fig:model-rank-pyr-lowratio-rank} shows that when pyruvate was the carbon source, the contrast of proteome allocation between biomass components in the low $\ratioabl$ condition is lessened compared to when glucose was the carbon source.
This may be explained by the higher $\ratioabl$ in this condition ($\ratioabl = 0.9$), compared to the $\ratioabl$ in the glucose-ammonium investigation ($\ratioabl = 0.7$).

\begin{figure}
  \centering
  \begin{subfigure}[t]{0.45\textwidth}
  \centering
    \includegraphics[width=\linewidth]{ec_grid_glc_amm_cosine}
    \caption{
    }
    \label{fig:model-noisy-glc-cosine}
  \end{subfigure}%
  \begin{subfigure}[t]{0.45\textwidth}
  \centering
    \includegraphics[width=\linewidth]{ec_grid_pyr_amm_cosine}
    \caption{
    }
    \label{fig:model-noisy-pyr-cosine}
  \end{subfigure}

  \begin{subfigure}[t]{0.45\textwidth}
  \centering
    \includegraphics[width=\linewidth]{ec_grid_glc_amm_kendall}
    \caption{
    }
    \label{fig:model-noisy-glc-kendall}
  \end{subfigure}%
  \begin{subfigure}[t]{0.45\textwidth}
  \centering
    \includegraphics[width=\linewidth]{ec_grid_pyr_amm_kendall}
    \caption{
    }
    \label{fig:model-noisy-pyr-kendall}
  \end{subfigure}

  \caption{
    Measures of similarity between enzyme usage fluxes in different nutrient conditions.
    Cosine distances between the between the vector of enzyme usage fluxes when protein is prioritised and the equivalent vector when carbohydrate is prioritised are shown \textbf{(\ref{fig:model-noisy-glc-cosine})} for glucose-ammonium conditions and \textbf{(\ref{fig:model-noisy-pyr-cosine})} pyruvate-ammonium conditions; smaller distances indicate a greater similarity between the two conditions.
    Means of the set of Kendall's $\tau$-$b$ rank correlation coefficients between the parallel case and each biomass-prioritised case are shown \textbf{(\ref{fig:model-noisy-glc-kendall})} for glucose-ammonium conditions and \textbf{(\ref{fig:model-noisy-pyr-kendall})} pyruvate-ammonium conditions; larger means indicate a greater similarity between parallel biosynthesis and individual biomass component synthesis.
    }
  \label{fig:model-noisy}
\end{figure}

The use of enzyme usage vectors to explain biomass synthesis scheduling strategies is called into question by the multiplicity of solutions in FBA.
Fig.\ \ref{fig:model-noisy} shows that the relationship between exchange rates and measures of similarity between enzyme usage fluxes when carbohydrate is prioritised and when protein is prioritised is not continuous.
This observation is likely explained by the multiplicity of solutions in FBA.
Namely, while FBA finds the optimal value of the objective function, there is no guarantee that the flux vector $\mathbf{v}$ (Eq.\ \ref{eq:model-fba-objective}) would be equivalent across linear programming solvers.
% I still don't like this sentence.
In sum, although choosing representative nutrient conditions leads to an attractive picture that may confirm the hypothesis of this section, the computational limitations of FBA may render any conclusion flawed.


\section{Conclusion}
\label{subsec:model-conclusion}

This chapter examines the choice between two strategies of scheduling synthesis of biomass components in the yeast cell: in sequence and in parallel.
% The cell synthesises its biomass components in sequence, re-allocating its proteome to enzymes that have roles related to the synthesis of each biomass component, as an adaptation to a limited proteome pool.
% However, changes in nutrient conditions may make parallel biosynthesis advantageous if the proteome allocation between biomass components become more similar.

Results show that ablation of components of the biomass reaction led to changes in proteome allocation to enzymes that are biologically relevant (section~\ref{sec:model-yeast8-pseudometabolites}) and provides a way to estimate the timescale of synthesis of biomass components (section~\ref{sec:model-timescale}).
%From these times, a $\ratioabl$ ratio was derived to assess whether sequential or parallel biosynthesis of biomass components is favoured.
These timescales suggest that sequential scheduling of biosynthesis saves times during growth and remains advantageous across deletion strains.

Results further show that within realistic growth rates, a smaller proteome pool led to a greater advantage of sequential biosynthesis (section~\ref{sec:model-pool}).
However, parallel scheduling of biosynthesis becomes advantageous in some nutrient conditions, such as when both carbon and nitrogen sources are limiting (section~\ref{subsec:model-grid}).
This may be explained by similar proteome allocation patterns across the synthesis of biomass components (section~\ref{subsec:model-rank}).

Conditions in which sequential biosynthesis is advantageous may explain why yeast cell show metabolic cycles in certain nutrient conditions.
Temporal scheduling of biosynthesis as an adaptation to a finite proteome may explain temporal gating of biosynthetic events in the yeast metabolic cycle.
Clustering of the synthesis of lipids and proteins in one group may explain cycling of storage lipids \parencite{campbellBuildingBlocksAre2020}, while clustering of carbohydrate, DNA, and RNA synthesis in another group to group together processes that require oxidative phosphorylation may explain respiratory cycles in the YMC \parencite{tuLogicYeastMetabolic2005}.
