%%
%% precntnt.tex - LaTeX2e thesis class
%%
%% Copyright (C) 2010-2021 Mathew Topper <damm_horse@yahoo.co.uk>
%%
%%
%%   ABOUT
%%
%% This is frontmatter prior to the contents for a Latex2e template which
%% corresponds to the regulations regarding layout of a thesis submitted
%% within the University of Edinburgh. 
%%
%% INPUT THIS FILE USING THE /makefrontmatter{} COMMAND OR THE FORMATTING
%% WON'T WORK PROPERLY

%%%% DEDICATION

\dedication{%
\begin{normalsize}To my family,\end{normalsize}\\[0.2cm]%
Mum and Dad, my brother Arun, and aunt Pinnapa.%
}

%%%% ABSTRACT
\abstract{
  The yeast metabolic cycle (YMC) is a biological rhythm in budding yeast (\emph{Saccharomyces cerevisiae}).
  It entails oscillations in the concentrations and redox states of intracellular metabolites, oscillations in transcript levels, temporal partitioning of biosynthesis, and, in chemostats, oscillations in oxygen consumption.
%  The YMC also responds to perturbations in nutrient media, generally being longer if nutrient quality is poor.
%  The YMC has also been found to gate phases of the cell division cycle and to share regulatory mechanisms with other biological rhythms such as the circadian oscillator.
  Most studies on the YMC have been based on chemostat experiments, and
  it is unclear whether YMCs arise from interactions between cells or are generated independently by each cell.
%  Furthermore, the molecular mechanism of the YMC is unclear.
  Thus, I aim to characterise the YMC in single cells and its response to nutrient and genetic perturbations.
  Specifically, I use microfluidics to trap and separate yeast cells, then record the time-dependent intensity of flavin autofluorescence, a component of the YMC.

  Single-cell microfluidics produces a large amount of time series data.
  Noisy and short time series produced from biological experiments restrict the computational tools that are useful for analysis.
  After considering the merits and caveats of such tools, I develop a method to filter time series, develop a machine learning model to classify whether time series are oscillatory, and develop a cross-correlation method to examine the periodicity of time series data.

  My experimental results show that yeast cells show oscillations in the fluorescence of flavins.
  Specifically, I show that in high glucose conditions, cells generate flavin oscillations asynchronously within a population, and these flavin oscillations couple with the cell division cycle.
%  These properties still hold when cells are cultivated in pyruvate, but the quality of oscillations is impaired in glucose-limiting conditions.
  I show that cells are able to individually reset the phase of their flavin oscillation, independently of the cell division cycle, in response to abrupt nutrient changes.
  I also show that deletion strains generate flavin oscillations that exhibit different behaviour from dissolved oxygen oscillations from chemostat conditions.

  Finally, I use flux balance analysis to address whether proteomic constraints in cellular metabolism mean that temporal partitioning of biosynthesis is advantageous for the yeast cell, and whether such partitioning explains the timing of the metabolic cycle.
  My result show that in certain conditions, it is advantageous for the cell to sequentially synthesise biomass components because doing so shortens the timescale of biomass synthesis.

  This thesis thus confirms autonomous generation of flavin oscillations, and suggests a model in which the YMC responds to nutrient conditions and subsequently entrains the cell division cycle.
  It also emphasises the possibility that subpopulations in the culture explain chemostat-based observations of the YMC.
  Furthermore, this thesis paves the way for using computational methods to analyse large datasets of oscillatory time series, which is useful for many fields of study beyond the YMC.
}


%%%% LAY SUMMARY
\summary{See the web page
\href{http://www.dcc.ac.uk/resources/how-guides/write-lay-summary}{"How to Write a Lay Summary"}
for a guide.}

%%%% ACKNOWLEDGEMENTS

\acknowledgements{%
  This work would not have been possible without the support of many people and organisations.

  I would like to thank the financial support provided by School of Biological Sciences, University of Edinburgh, and the Edinburgh Global Scholarship.

  I would like to express my deepest gratitude towards my supervisors Dr Diego Oyarz\'{u}n and Prof Peter Swain for the unique opportunity of joining two research teams, and for creating such a great environment for me to learn and work in Edinburgh.
  I am also very grateful your advice, patience, and mental support -- especially in the very isolating days of the COVID-19 pandemic.

  I am most grateful for Dr Kevin Correia for his mentorship in the early parts of my PhD: especially teaching me how to me ask questions, bringing in laboratory best practices from his experience, and even philosophical insights in science and life.

  I am very grateful for the main software developers of the Swain lab: Dr Julian Pietch, Dr Ivan Clark, Dr Diane Adjavon, and Dr Al\'{a}n Mu\~{n}oz Gonzalez.
  I would like to thank Julian in particular for his patience as I grasped the ropes of the old software pipeline.
  Thank you, Diane and Al\'{a}n, for including me in your very ambitious project to re-write the pipeline -- this was an immense learning experience for me.
  Al\'{a}n -- we've been through thick and thin, and I'm most grateful to have you as a partner in crime (?), and for \texttt{doom emacs} too.

  I am very grateful for Dr Ivan Clark and Iseabail Farquhar for imparting their wet-lab expertise and their patience during my experiments.
  Ish, I'd like to thank you for your patience and perseverance training me CRISPR during the chaotic times of the COVID-19 lockdown.

  I would also like to thank my other colleagues at the Swain Lab: Dr Fran\c{c}ois El-Daher, Dr Julien Hurbain, Dr Nahuel Manzanaro Moreno, Dr Yu Huo, and Sof\'{i}a Esteban Serna.

  I would like to thank my colleagues at the Biomolecular Control Group for creating a fantastic social environment and continuously sparking ideas.
  I am grateful for Dr Vanessa Smer-Barreto and Dr Babita Verma for their mentorship and feedback on my research.
  I would like to thank Dr Mona Tonn and Denise Thiel for our weekly movie nights and continued moral support.
  And I would like to thank Evangelos Marios-Nikolados for a great collaboration opportunity, and (too) regular pub nights.
  Let's drink to health!

  I would like to thank my friends from university -- Christos Nicolaou, Esther Ng, Patawee Jintana, among others -- for their moral support.
  I would like to extend special thanks to my best friends Jessica Wang and Maetavee Asavabhokhin for their unwavering emotional and moral support during all ups and downs.
  And I would also like to thank my high school friends, scattered across the globe, Attawit Chaiyaroj, Paloch Vasudhara, and Dr Pravarit Athiprayoon for their support too.

  Last but not least, I would like to thank my family.
  I express my deepest gratitude for my parents who have taught me the wonder of science, the importance of education, and the value of working hard.
  None of this would have been possible without you.
  Aunt Pinnapa, I now have a PhD!
  And lastly, I thank my brother Arun Wongprommoon and cousin Pornsom Ruengpirasiri for their continuous support in this long journey.
}

%%%% DECLARATION

%% Use a custom declaration

% \declaration{I did it.}

%% Use the standard regulation declaration. Enter your
%% name for the signature line.

\standarddeclaration{Arin Wongprommoon}

%%% Local Variables:
%%% mode: latex
%%% TeX-master: "../thesis"
%%% End:
